# (PART\*) Standard workflow {.unnumbered}

# Setup {#setup}

```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/start.png", auto_pdf = TRUE)
```

## Workshop directory & data

Prior to starting analysis we will create a working directory. The directory `~/bacterial_assembly/standard_workflow` will contain all the inputs and outputs of our analyses.

```{bash eval=FALSE}
#Make directory
mkdir -p ~/bacterial_assembly/standard_workflow
#Move into it
cd ~/bacterial_assembly/standard_workflow
```

During the workflow we will keep a tidy directory structure outputting the results form different tools into their respective directories.

For the standard workflow we will be using pre QC'd PacBio reads of an *Escherichia coli* genome. Create a directory and create soft links (i.e. a shortcut) of the read data.

```{bash eval=FALSE}
#Create directory
mkdir ecoli_reads
#Softlink of read data
ln -s /pub39/tea/nsc006/NEOF/bact_assembly/data/ecoli.fastq ecoli_reads
```

## Conda environments

```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
```

`Conda` is a very useful tool for bioinformaticians. It allows for the creation of virtual environments so you can install different programs. This is generally easier than manually installing programs for 2 main reasons:

-   Usually `Conda` will install all the dependencies a program needs.
    -   Prior to `Conda` it could take more than a day to install a program and all its dependencies plus the dependencies of the dependencies etc.
-   Different programs may need different versions of the same program as a dependency (e.g. One programs may need Python2 whilst another needs Python3). Therefore installing one can break the other. Having separate `Conda` environments (virtual environments) can isolate clashing programs.

**Analogy**: You can think of programs as food and environments as food storage units. You could try to shove all your food into one giant cold room but most of your food will either be too cold or too warm. Instead it would be better to have different types of food in different environments as outlined in the below table.

| Storage environment | Food examples                      |
|---------------------|------------------------------------|
| Fridge              | Fresh vegetables, fresh meat, etc. |
| Freezer             | Frozen meat, ice, etc.             |
| Pantry              | Canned food, jarred food, etc.     |
| Cellar              | Wine.                              |
| Fruit bowl          | Fruit.                             |

As we will be using many different programs we will be using different `Conda` environments. To activate these environments you will be using `use` scripts that will activate the relevant `Conda` environment belonging to user `nsc006` (Matthew Gemmell).

Preferably during the course of this workflow you will have a terminal open for each `Conda` environment. It is important to make sure you are in the correct terminal/environment for each chapter. 

You can see what environment a terminal currently has activated by looking at the command prompt. Below we can see the command prompt shows we the terminal is in the environment called `bacterial_assembly`. The name of the currenlty activated `Conda` environment will always be in `()` before the login/user name across all systems.

```{r, fig.align = 'center',out.width= '60%', echo=FALSE }
knitr::include_graphics(path = "figures/conda_bacterial_assembly_terminal.png", auto_pdf = TRUE)
```

For your own future analyses you would use your own `Conda`. If you are interested please see the `Conda` and `Mamba` links in the [Next steps](#nextsteps) section of the Appendix.