# Genome assembly assessment {#genomeassemblyassessment}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/inspect.png", auto_pdf = TRUE)
```

After carrying out an assembly it is always advised to assess its quality. We will carry this out with three tools:

- `QUAST` - Gives assembly stats related to contiguity.
- `BUSCO` - Estimates completeness of a genome.
- `CheckM` - Estimates completeness and contamination of a genome.

## Assessment: Conda environnment & directory
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
``` 

Open a new terminal (right click on the main screen, choose `Applications` -> `Shell` -> `bash`). 

We will use the `bacterial_genome_assessment` `conda` environment for our genome assessment tools.

```{bash eval=FALSE}
chos 8
. usebacterialgenomeassessment
```

Ensure you are in the the `standard_workflow` directory for this new terminal and environment.

```{bash eval=FALSE}
cd ~/bacterial_assembly/standard_workflow
```

## QUAST
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/measuring_tape.png", auto_pdf = TRUE)
``` 

`QUAST` is the most used tool to assess genomes. It gives various statistics related to the contiguity of the assembly. Unless stated these statistics are based on contigs that are >=500bp.

Create an output directory and run `QUAST`.

```{bash eval=FALSE}
#Make output directory
mkdir -p quast/redbean
#Run QUAST
quast -o quast/redbean redbean_assembly/ecoli.ctg.fa
```

__Parameters__

- `-o` : The output directory.
- The input file/s are at the end of the command without a flag (e.g. `-o`).
   - __Note__ : `QUAST` can be run on multiple assemblies giving a report which contains info on all the input assemblies.

`QUAST` produces reports in various formats. We will look at the html report.

```{bash eval=FALSE}
firefox quast/redbean/report.html
```

Inspect the table and interact with the interactive plots.

Statistics such as length, # contigs (number of contigs), and GC (%) are self explanatory. Two definitions that you may not be aware of are N50/N75 and L50/L75. 

To calculate N50 and L50:

1. Order the contigs from largest to smallest.
2. Find the point that is 50% the length of the assembly starting from the start of the largest contig going toward the smallest contig.
3. Determine which contig this point belongs to.
4. The N50 is equal to the length of the contig.
5. The L50 is equal to the number of contigs that is >= in length to the N50.

To find the N75 & L75 change the 50% to 75% in point 2. You can use any number between 1-100 for the Nx or Lx.

Additionally, below is a visual demonstration of Nx and Lx.

```{r, fig.align = 'center',out.width= '60%', echo=FALSE }
knitr::include_graphics(path = "figures/n50_n90.png", auto_pdf = TRUE)
``` 

__Questions__

- How many total contigs are there?
- Approximately how much bigger is the total assembly compared to the largest contigs (to the nearest 1k)?
- Does the GC% match what is know about _E.coli_? (You may need to google)
- Why is the N50 and N75 the same?

Overall these contiguity stats are good. It is not a one contig assembly, which is what we would aim for a single chromosome genome, but it is close. It mainly consists of one large contig with 3 smaller contigs.

Some things you will want to look out for in future projects:

- High number of contigs, with small N50: This may indicate the genome assembly is very uncontiguous and so the assembler had issues with assembly. 
   - It is possible the genome is a particularly difficult genome to resolve.
   - The input data could be poor. Examples include:
      - Low quality.
      - Low coverage.
      - Average coverage could be good but it is possible areas of the genome have low coverage so these areas could not be assembled well.
- Length is not correct: Could be caused by several reasons:
   - Parts of the genome are duplicated in the assembly causing the assembly to be too long. This can be caused by parts where a consensus could not be reached effectively.
   - A short assembly can indicate that parts of the genome are missing.
   - A drastically increased length can indicate the presence of contamination from another organism. The GC (%) can give a clue if this may be the case.
   
Further assessment can elucidate if there are any issues and what the issue may be.

## BUSCO
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/universe.png", auto_pdf = TRUE)
``` 

`BUSCO` stands for Benchmarking Universal Single-Copy Orthologs. The tool is used to estimate the completeness of genome assemblies by determining how many BUSCOs are present. 

Prior to running `BUSCO` we will create an output directory for the `BUSCO` analysis.

```{bash eval=FALSE}
mkdir -p busco/redbean
```

BUSCOs are genes that are:

- __Universal__: The genes are present in all organisms of a specific lineage.
- __Single-Copy__: The genes are only found as one copy.

`BUSCO` has many different lineage datasets. These are datasets of the different groups of BUSCOs found in various different lineages. Carry out the below command to see all these different datasets.

```{bash eval=FALSE}
busco --list-datasets | less -S
```

You will notice that the lineage datasets are at various taxonomic levels going from kingdom (bacteria, archaea, & eukaryota) down to order (rhizobiales, burkholderiales etc.). It is always good to be as specific as you can to your organisms lineage. A more specific lineage will have more BUSCOs that a more generic one.

__Note__: All datasets end with `_odb10` which tands for OrthoDB version 10 (https://www.orthodb.org/).

Looking through the choices (and using NCBI taxonomy) what lineage dataset would you use for the following genera?

- __Psychrobacter__ (Taxonomy ID: 497)
- __Iamia__ (Taxonomy ID: 467975)
- __Phycisphaera__ (Taxonomy ID: 666508)
- __Atribacter__ (Taxonomy ID: 2847777)

For our `E.coli` we will choose the `enterobacterales_odb10` as our lineage dataset. With that we can run `BUSCO` on our genome assembly.

```{bash eval=FALSE}
busco \
-i redbean_assembly/ecoli.ctg.fa \
-l enterobacterales_odb10 \
-m geno \
-o ecoli \
--out_path busco/redbean
```

__Parameter choice__

- `-i` : The input genome assembly in fasat format.
- `-l` : The lineage dataset to be used. This has to end with the `_odb10` part. This will download the dataset (if it does not exist) into a directory called `busco_downloads` in the current directory. The directory will be created if it doesn't exist.
- `-m` : The assessment mode.
   - `geno` = genome assemblies (DNA)
   - `tran` = transcriptome assemblies (DNA)
   - `prot` = annotated gene sets (protein)
- `-o` : The run name. Output folders and files will be labelled with this name.
- `--out_path` : Path where the output directory will be created.

There are many output files in the directory `busco/redbean/ecoli`. A full list of the output can be found in the following link: https://busco.ezlab.org/busco_userguide.html#outputs

I ignore most of the output most of the time, instead only looking at the short summary file.

```{bash eval=FALSE}
less busco/redbean/ecoli/short_summary.specific.enterobacterales_obd10.ecoli.txt
```

This provides us with 6 values. These values are presented in terms of percentage (along the top line of the Results) and total numbers (the next 6 lines). The values represent:

- __Complete BUSCOs__ : The number of complete BUSCOs discovered. The closer this value is to 100% the better. This number is the sum of the next two values.
- __Complete and single-copy BUSCOs__: The number of BUSCOs found as complete and single-copy. The closer this number is to 100% the better. A good assembly will generally be >90%.
- __Complete and duplicated BUSCOs__: Complete BUSCOs that have been found two or more times within the assembly. This indicates there are duplicated areas of the genome. Preferably we would like this number to be low. 
- __Fragmented BUSCOs__: This could indicate that genes that only partially present within the genome. This could be due to misassmebly.
- __Missing BUSCOs__: These are BUSCOs that could not be found in the genome assembly. This could indicate parts of the genome are missing in the assembly.
- __Total BUSCO groups searched__: This represents the total number of BUSCOs within lineage dataset used for analysis.

For more info on interpreting the results please see the following link: https://busco.ezlab.org/busco_userguide.html#interpreting-the-results

__Questions__

- Ignoring total BUSCOs, what is the highest value? Do you want this value to be the highest?
- Are these values good? What may the stats indicate?

The `BUSCO` stats are ok. There are a bit more fragmented and missing BUSCOs than we would like. We'll try to improve this with polishing. We could also try to get some better data (maybe some Illumina data) or try a different assembler. However, as this is a workshop we will continue on.

## CheckM
```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/CheckM.png", auto_pdf = TRUE)
``` 

`CheckM` can also be used to estimate the completeness of a bacterial genome, however I prefer `BUSCO` for this. Instead, we will use `CheckM` to estimate contamination within our assembly.

First create an output directory.

```{bash eval=FALSE}
mkdir -p checkm/redbean/ecoli
```

`CheckM` has many different commands. We will use its typical workflow which is conveniently wrapped into one command.

```{bash eval=FALSE}
checkm lineage_wf --tab_table -f checkm/ecoli_checkm.tsv -t 8 -x ctg.fa redbean_assembly/ checkm/redbean/ecoli
```

__Parameters__

- `--tab_table` : Creates a `.tsv` file (tab seperated value) that acts a summary of results. This is the one file we will inspect.
- `-f` : Path for output summary `.tsv` file.
- `-t` : Number of threads to be used.
- `-x` : Suffix of genome assembly files to be analysed. The command will carry out analysis on all the files with this suffix in the __bin directory__.
- The second last option (which is flagless) is the __bin directory__. The command will carry out the workflow on each file with the suffix as stated in the `-x` parameter.
- The last option (which is flagless) is the output directory.

Ignore all the other files and view the summary files.

```{bash eval=FALSE}
less -S checkm/redbean/ecoli/ecoli_checkm.tsv
```

There are quite a few statistics but we are only interested in the  `Contamination` column (you may need to use the right key to move to it). This value is a percentage. Is it a large value?

Thankfully not, it is less than 1%. If this was larger than 1% we may need to worry. If that was the case we could try to determine if it is real contamination and remove it if so. There is info on how to crry this out in the supplemental bookdown.

If you are interested in the other values please check: https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics

## Assessment recap
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/recap.png", auto_pdf = TRUE)
``` 

Our assembly looks good. Good contiguity figures from `QUAST`, and `CheckM` shows low signs, if any, contamination. The `BUSCO` values are not brilliant with some fragmented and missing BUSCOs.

In real analysis I would try another assembler to see if that would improve the BUSCO situation. For the purpose of time we will continue and I will let you explore other assemblers in your own time.