# (APPENDIX) Appendix {-}

# Next steps{#nextsteps}
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/step.png", auto_pdf = TRUE)
``` 

- `Conda` and `Mamba` are invaluable tools to install programs and create virtual environments.
- Benchmarks of long-read assemblers paper.
   - https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6966772/




# Manuals{#manuals}
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/manual.png", auto_pdf = TRUE)
``` 

Conda: https://conda.io/projects/conda/en/latest/user-guide/getting-started.html

Mamba: https://github.com/mamba-org/mamba

Redbean (wtdbg2): https://github.com/ruanjue/wtdbg2

Bandage: https://rrwick.github.io/Bandage/

GFA spec: http://gfa-spec.github.io/GFA-spec/GFA1.html

Canu: https://github.com/marbl/canu

HGAP: https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP

Flye: https://github.com/fenderglass/Flye

Raven: https://github.com/lbcb-sci/raven

NECAT: https://github.com/xiaochuanle/NECAT

Shasta: https://github.com/chanzuckerberg/shasta

QUAST: http://quast.sourceforge.net/quast

BUSCO: https://busco.ezlab.org/

CheckM: https://github.com/Ecogenomics/CheckM/wiki

BWA: https://github.com/lh3/bwa

SAM: https://samtools.github.io/hts-specs/SAMv1.pdf

RACON: https://github.com/isovic/racon

MUMmer: http://mummer.sourceforge.net/

dnadiff: https://github.com/mummer4/mummer/blob/master/docs/dnadiff.README

Pilon: https://github.com/broadinstitute/pilon/wiki

PacBio's GenomicConsensus: https://github.com/broadinstitute/pilon/wiki

HyPo: https://github.com/kensung-lab/hypo

Medaka: https://github.com/nanoporetech/medaka

NextPolish: https://github.com/Nextomics/NextPolish






FastQC: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

MultiQC: https://multiqc.info/

Trim Galore: https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/

Bowtie2: http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml

samtools: http://www.htslib.org/

BBTools: https://jgi.doe.gov/data-and-tools/bbtools/

Kraken2: https://github.com/DerrickWood/kraken2/wiki/Manual

Krona: https://github.com/marbl/Krona/wiki/KronaTools

Bracken: https://ccb.jhu.edu/software/bracken/index.shtml?t=manual

LEfSe: https://huttenhower.sph.harvard.edu/lefse/

HUMAnN2: https://github.com/biobakery/biobakery/wiki/humann2

MetaPhlAn2: https://huttenhower.sph.harvard.edu/metaphlan2/

Biobakery: https://github.com/biobakery/biobakery

MegaHit: https://github.com/voutcn/megahit

BWA: https://github.com/lh3/bwa

minimap2: https://github.com/lh3/minimap2

MetaBAT2: https://bitbucket.org/berkeleylab/metabat/src/master/

CheckM: https://github.com/Ecogenomics/CheckM/wiki

PhyloPhlAn: https://github.com/biobakery/phylophlan/wiki

MetaWRAP: https://github.com/bxlab/metaWRAP

Pokka: https://github.com/tseemann/prokka

MinPath: https://github.com/mgtools/MinPath/blob/master/readme

MetaCyc: https://metacyc.org/

# Obtaining Read Data

https://github.com/PacificBiosciences/DevNet/wiki/8-plex-Ecoli-Multiplexed-Microbial-Assembly

