---
title: "Bacterial genome assembly and annotation"
author: "Matthew R. Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
favicon: figures/NEOF_favicon.png
description: NEOF book for the Bacterial genome assembly and annotation workshop
cover-image: "figures/NEOF.png"
---

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
```

# (PART\*) Start {-}
# Introduction

```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/map.png", auto_pdf = TRUE)
```

This practical session will run you through a standard bacterial genome assembly and annotation workflow using PacBio data. The topics covered are:

- [Setup](#setup)
- [Overview](#overview)
- [Genome assembly with `Redbean` (`wtdbg2`)](#redbeanassembly)
- [Assembly assessment](#genomeassemblyassessment)
- [Polishing with `arrow`](#polishing)
- [Circularisation with `Circlator`](#Circlator)
- [Annotation with `Prokka`](#annotation)
- [Final report generation with `MultiQC`](#finalreport)

The aim of this practical is to produce a genome assembly of _Eschrichia coli_, annotate it, and produce a report with the assembly's stats. Read QC will be skipped as this has been covered in a previous NEOF workshop.

Commands are in the following font, colour, and box.They should be run in the command line.

```{bash eval=FALSE}
echo "This is a command example" 
```

<!--chapter:end:01-Bacterial_genome_assembly_and_annotation.Rmd-->

# Cluster Introduction
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/cluster.png", auto_pdf = TRUE)
``` 

## Logon instructions
For this workshop we will be using Virtual Network Computing (VNC). Connect to the VNC with a browser by using the webVNC link you were sent.

You will now be in a logged-in Linux VNC desktop. You will see something as below (there may be only one terminal which is fine). If you do not see something similar please ask for assistance.

```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/logon_pic.png", auto_pdf = TRUE)
``` 

If the VNC is taking up too much/little space of your browser you can use the zoom of your browser to adjust the size. Ensure you can see one whole terminal.

These instructions will not work outside of this workshop. If you would like to install your own Linux OS on your desktop or laptop we would recommend Ubuntu. 

The following link is a guide to install Ubuntu:  
https://www.ubuntu.com/download/desktop/install-ubuntu-desktop.  
If you use a USB you need to create a bootable USB stick. The following link will assist:  
https://www.ubuntu.com/download/desktop/create-a-usb-stick-on-windows 

## The Terminal Window
In our case the terminal window looks like the picture below. We are using the terminal window as our shell to interpret our commands to the kernel. Depending on your system and preferences it may look different.
```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/terminal_window.png", auto_pdf = TRUE)
``` 

Already there is useful information for us on the terminal window.

- __nsc065__: This is the login name, also known as the username. In this case nsc065 is a demonstrator's account. Your screen should show a different account name which will be your username for the Linux machine/cluster you are logged into.
- __gauss03__: This is the machine name the user is logged into.
- __\~__: This represents the current directory of the user, or the directory a command was run in. In the Linux OS and others __'~'__ is a shortcut to the user's home directory.
- Everything after the __'$'__ is where commands are typed into the terminal. This is also referred to as the command line.

__To open a new terminal window__, right click on the main screen, choose `Applications` -> `Shell` -> `bash`

<!--chapter:end:02-Cluster_Introduction.Rmd-->

# (PART\*) Standard workflow {.unnumbered}

# Setup {#setup}

```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/start.png", auto_pdf = TRUE)
```

## Workshop directory & data

Prior to starting analysis we will create a working directory. The directory `~/bacterial_assembly/standard_workflow` will contain all the inputs and outputs of our analyses.

```{bash eval=FALSE}
#Make directory
mkdir -p ~/bacterial_assembly/standard_workflow
#Move into it
cd ~/bacterial_assembly/standard_workflow
```

During the workflow we will keep a tidy directory structure outputting the results form different tools into their respective directories.

For the standard workflow we will be using pre QC'd PacBio reads of an *Escherichia coli* genome. Create a directory and create a soft link (i.e. a shortcut) of the read data.

```{bash eval=FALSE}
#Create directory
mkdir ecoli_reads
#Softlink of read data
ln -s /pub39/tea/nsc006/NEOF/bact_assembly/data/ecoli.fastq ecoli_reads
```

## Conda environments

```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
```

`Conda` is a very useful tool for bioinformaticians. It allows for the creation of virtual environments so you can install different programs. This is generally easier than manually installing programs for 2 main reasons:

-   Usually `Conda` will install all the dependencies a program needs.
    -   Prior to `Conda` it could take more than a day to install a program and all its dependencies plus the dependencies of the dependencies etc.
-   Different programs may need different versions of the same program as a dependency (e.g. Python version 2 or version 3). Therefore installing one can break the other. Having separate `Conda` environments (virtual environments) can isolate clashing programs.

**Analogy**: You can think of programs as food and environments as food storage units. You could try to shove all your food into one giant cold room but most of your food will either be too cold or too warm. Instead it would be better to have different types of food in different compartments as outlined in the below table.

| Storage environment | Food examples                      |
|---------------------|------------------------------------|
| Fridge              | Fresh vegetables, fresh meat, etc. |
| Freezer             | Frozen meat, ice, etc.             |
| Pantry              | Canned food, jarred food, etc.     |
| Cellar              | Wine.                              |
| Fruit bowl          | Fruit.                             |

As we will be using many different programs we will be using different `Conda` environments. To activate these environments you will be using `use` scripts that will activate the relevant `Conda` environment belonging to user `nsc006` (Matthew Gemmell).

Preferably during the course of this workflow you will have a terminal open for each `Conda` environment. It is important to make sure you are in the correct terminal/environment for each chapter. 

You can see what environment a terminal currently has activated by looking at the command prompt. Below we can see the command prompt shows we are in the environment called `bacterial_assembly` due the environment name being in the `()` before your login/user name. This is consistent with `Conda` in all systems.

```{r, fig.align = 'center',out.width= '60%', echo=FALSE }
knitr::include_graphics(path = "figures/conda_bacterial_assembly_terminal.png", auto_pdf = TRUE)
```

For your own future analysis you would use your own `Conda`. If you are interested please see the `Conda` and `Mamba` links in the [Next steps](#nextsteps) section of the Appendix.

<!--chapter:end:03-Start.Rmd-->

# Overview {#overview}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/overview.png", auto_pdf = TRUE)
``` 

<!--chapter:end:04-Overview.Rmd-->

# Genome Assembly {#redbeanassembly}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/redbean.png", auto_pdf = TRUE)
``` 

(bit of a spiel about assembly)

## Redbean

Redbean (AKA wtdbg2)


### Redbean: Conda environnment & directory
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
``` 

We will use the `bacterial_assembly` conda environment for our Redbean assembly.

```{bash eval=FALSE}
. usebacterialassembly
```

### Redbean: Assemble long reads

```{bash eval=FALSE}
wtdbg2 -x rs -g 4.6m -i ecoli_reads/ecoli.fastq -o redbean_assembly/ecoli -t 8
```

### Redbean: Derive consensus

```{bash eval=FALSE}
wtpoa-cns -t 8 -i redbean_assembly/ecoli.ctg.lay.gz -fo redbean_assembly/ecoli.ctg.fa
```

### Redbean: Bandage Visualisation

(for interest about assembly graph)
(view assembly graphs, why consensus is derived)

`gunzip` files
```{bash eval=FALSE}
gunzip redbean_assembly/ecoli.1.dot.gz
gunzip redbean_assembly/ecoli.2.dot.gz
gunzip redbean_assembly/ecoli.3.dot.gz
```

Convert `.dot` file into `.gfa` file required for `Bandage`.

The script `wtdbg2-dot2gfa.pl` is found in the scripts directory of `wtdbg2` which can be found at: https://github.com/ruanjue/wtdbg2/tree/master/scripts. Unfortunately this is not installed when you install `wtdbg2` via `conda`.
```{bash eval=FALSE}
wtdbg2-dot2gfa.pl redbean_assembly/ecoli.1.dot > redbean_assembly/ecoli.1.dot.gfa
wtdbg2-dot2gfa.pl redbean_assembly/ecoli.2.dot > redbean_assembly/ecoli.2.dot.gfa
wtdbg2-dot2gfa.pl redbean_assembly/ecoli.3.dot > redbean_assembly/ecoli.3.dot.gfa
```

Open up `Bandage` GUI.
```{bash eval=FALSE}
Bandage
```

File -> Load Graph -> 
Graph Drawing -> Draw graph

## Other long read assemblers

<!--chapter:end:05-Assembly.Rmd-->

# Genome assembly assessment {#genomeassemblyassessment}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/inspect.png", auto_pdf = TRUE)
```


## Assessment: Conda environnment & directory
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
``` 

We will use the `bacterial_genome_assessment` `conda` environment for our `Redbean` assembly.

```{bash eval=FALSE}
chos 8
. usebacterialassembly
```

```{bash eval=FALSE}
cd ~/bacterial_assembly/standard_workflow
```

## QUAST

```{bash eval=FALSE}
mkdir -p quast/redbean
quast -o quast/redbean redbean_assembly/ecoli.ctg.fa
```

```{bash eval=FALSE}
firefox quast/redbean/report.html
```

## BUSCO

```{bash eval=FALSE}
mkdir -p busco/redbean
cd busco/redbean
```

```{bash eval=FALSE}
busco --list-datasets | less -S
```

```{bash eval=FALSE}
busco -i ../../redbean_assembly/ecoli.ctg.fa -l enterobacterales_odb10 -m geno -o ecoli
```

```{bash eval=FALSE}
less ecoli/short_summary.specific.enterobacterales_obd10.ecoli.txt
```

## CheckM

```{bash eval=FALSE}
cd ~/bacterial_assembly/standard_workflow
mkdir -p checkm/redbean/ecoli
```

```{bash eval=FALSE}
checkm lineage_wf --tab_table -f checkm/ecoli_checkm.tsv -t 8 -x ctg.fa redbean_assembly/ checkm/redbean/ecoli
```

```{bash eval=FALSE}
less -S checkm/redbean/ecoli/ecoli_checkm.tsv
```

## MultiQC

```{bash eval=FALSE}
mkdir -p multiqc/redbean/ecoli/input
ln -s quast/redbean/report.tsv multiqc/redbean/ecoli/input
ln -s busco/redbean/ecoli/short_summary.specific.enterobacterales_odb10.ecoli.txt\
multiqc/redbean/ecoli/input
```

```{bash eval=FALSE}
multiqc --outdir multiqc/redbean/ecoli/input multiqc/redbean/ecoli
```

```{bash eval=FALSE}
firefox multiqc/redbean/ecoli/multiqc_report.html
```


MultiQC modules: https://multiqc.info/docs/#multiqc-modules

<!--chapter:end:06-Genome_assessment.Rmd-->

# Polishing {#polishing}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/polishing.png", auto_pdf = TRUE)
``` 

<!--chapter:end:07-Polishing.Rmd-->

# Circlator {#Circlator}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/circlator.png", auto_pdf = TRUE)
``` 

<!--chapter:end:08-Circlator.Rmd-->

# Annotation {#annotation}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/quokka.png", auto_pdf = TRUE)
``` 

<!--chapter:end:09-Annotation.Rmd-->

# Final report {#finalreport}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/quokka.png", auto_pdf = TRUE)
``` 

<!--chapter:end:10-Final_report.Rmd-->

# (APPENDIX) Appendix {-}

# Next steps{#nextsteps}
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/step.png", auto_pdf = TRUE)
``` 

Below are some good links to start with before carrying out your own projects.

(Conda and Mamba)

https://github.com/mamba-org/mamba


# Manuals
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/manual.png", auto_pdf = TRUE)
``` 

Conda: https://conda.io/projects/conda/en/latest/user-guide/getting-started.html

FastQC: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

MultiQC: https://multiqc.info/

Trim Galore: https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/

Bowtie2: http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml

samtools: http://www.htslib.org/

BBTools: https://jgi.doe.gov/data-and-tools/bbtools/

Kraken2: https://github.com/DerrickWood/kraken2/wiki/Manual

Krona: https://github.com/marbl/Krona/wiki/KronaTools

Bracken: https://ccb.jhu.edu/software/bracken/index.shtml?t=manual

LEfSe: https://huttenhower.sph.harvard.edu/lefse/

HUMAnN2: https://github.com/biobakery/biobakery/wiki/humann2

MetaPhlAn2: https://huttenhower.sph.harvard.edu/metaphlan2/

Biobakery: https://github.com/biobakery/biobakery

MegaHit: https://github.com/voutcn/megahit

BWA: https://github.com/lh3/bwa

minimap2: https://github.com/lh3/minimap2

MetaBAT2: https://bitbucket.org/berkeleylab/metabat/src/master/

CheckM: https://github.com/Ecogenomics/CheckM/wiki

PhyloPhlAn: https://github.com/biobakery/phylophlan/wiki

MetaWRAP: https://github.com/bxlab/metaWRAP

Pokka: https://github.com/tseemann/prokka

MinPath: https://github.com/mgtools/MinPath/blob/master/readme

MetaCyc: https://metacyc.org/

# Obtaining Read Data

https://github.com/PacificBiosciences/DevNet/wiki/8-plex-Ecoli-Multiplexed-Microbial-Assembly


<!--chapter:end:11-Appendix.Rmd-->

