<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Circularisation | Bacterial genome assembly and annotation</title>
  <meta name="description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Circularisation | Bacterial genome assembly and annotation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Circularisation | Bacterial genome assembly and annotation" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2021-08-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="07-Polishing.html"/>
<link rel="next" href="09-Annotation.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation</a></li>

<li class="divider"></li>
<li class="part"><span><b>Start</b></span></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_assembly_and_annotation.html"><a href="01-Bacterial_genome_assembly_and_annotation.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Standard workflow</b></span></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Overview.html"><a href="04-Overview.html#overview-bacterial-genome-assembly"><i class="fa fa-check"></i><b>4.1</b> Overview: Bacterial genome assembly</a></li>
<li class="chapter" data-level="4.2" data-path="04-Overview.html"><a href="04-Overview.html#overview-long-read-data"><i class="fa fa-check"></i><b>4.2</b> Overview: Long read data</a></li>
<li class="chapter" data-level="4.3" data-path="04-Overview.html"><a href="04-Overview.html#overview-read-qc"><i class="fa fa-check"></i><b>4.3</b> Overview: Read QC</a></li>
<li class="chapter" data-level="4.4" data-path="04-Overview.html"><a href="04-Overview.html#overview-assembly"><i class="fa fa-check"></i><b>4.4</b> Overview: Assembly</a></li>
<li class="chapter" data-level="4.5" data-path="04-Overview.html"><a href="04-Overview.html#overview-assembly-assessment"><i class="fa fa-check"></i><b>4.5</b> Overview: Assembly assessment</a></li>
<li class="chapter" data-level="4.6" data-path="04-Overview.html"><a href="04-Overview.html#overview-polishing"><i class="fa fa-check"></i><b>4.6</b> Overview: Polishing</a></li>
<li class="chapter" data-level="4.7" data-path="04-Overview.html"><a href="04-Overview.html#overview-circlator"><i class="fa fa-check"></i><b>4.7</b> Overview: Circlator</a></li>
<li class="chapter" data-level="4.8" data-path="04-Overview.html"><a href="04-Overview.html#overview-annotation"><i class="fa fa-check"></i><b>4.8</b> Overview: Annotation</a></li>
<li class="chapter" data-level="4.9" data-path="04-Overview.html"><a href="04-Overview.html#overview-final-report"><i class="fa fa-check"></i><b>4.9</b> Overview: Final report</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Assembly.html"><a href="05-Assembly.html"><i class="fa fa-check"></i><b>5</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean"><i class="fa fa-check"></i><b>5.1</b> Redbean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbeanassemblyconda"><i class="fa fa-check"></i><b>5.1.1</b> Redbean: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-assemble-long-reads"><i class="fa fa-check"></i><b>5.1.2</b> Redbean: Assemble long reads</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-derive-consensus"><i class="fa fa-check"></i><b>5.1.3</b> Redbean: Derive consensus</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Assembly.html"><a href="05-Assembly.html#bandage-visualisation"><i class="fa fa-check"></i><b>5.2</b> Bandage Visualisation</a></li>
<li class="chapter" data-level="5.3" data-path="05-Assembly.html"><a href="05-Assembly.html#other-long-read-assemblers"><i class="fa fa-check"></i><b>5.3</b> Other long read assemblers</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html"><i class="fa fa-check"></i><b>6</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-conda-environnment-directory"><i class="fa fa-check"></i><b>6.1</b> Assessment: Conda environnment &amp; directory</a></li>
<li class="chapter" data-level="6.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast"><i class="fa fa-check"></i><b>6.2</b> QUAST</a></li>
<li class="chapter" data-level="6.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>6.3</b> BUSCO</a></li>
<li class="chapter" data-level="6.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#checkm"><i class="fa fa-check"></i><b>6.4</b> CheckM</a></li>
<li class="chapter" data-level="6.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>6.5</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Polishing.html"><a href="07-Polishing.html"><i class="fa fa-check"></i><b>7</b> Polishing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon"><i class="fa fa-check"></i><b>7.1</b> Racon</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-conda-environment-directory"><i class="fa fa-check"></i><b>7.1.1</b> Racon: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-index-assembly"><i class="fa fa-check"></i><b>7.1.2</b> Racon: Index assembly</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-read-alignment-to-assembly"><i class="fa fa-check"></i><b>7.1.3</b> Racon: Read alignment to assembly</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing"><i class="fa fa-check"></i><b>7.1.4</b> Racon: Polishing</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing-report"><i class="fa fa-check"></i><b>7.1.5</b> Racon: Polishing report</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-second-iteration"><i class="fa fa-check"></i><b>7.1.6</b> Racon: Second iteration</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Polishing.html"><a href="07-Polishing.html#other-polishers"><i class="fa fa-check"></i><b>7.2</b> Other polishers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Circlator.html"><a href="08-Circlator.html"><i class="fa fa-check"></i><b>8</b> Circularisation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator"><i class="fa fa-check"></i><b>8.1</b> Circlator</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-conda-environment-directory"><i class="fa fa-check"></i><b>8.1.1</b> Circlator: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-read-correction"><i class="fa fa-check"></i><b>8.1.2</b> Circlator: Read correction</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-circularisation"><i class="fa fa-check"></i><b>8.1.3</b> Circlator: Circularisation</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-assessment-polishing"><i class="fa fa-check"></i><b>8.1.4</b> Circlator: Assessment &amp; Polishing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Annotation.html"><a href="09-Annotation.html"><i class="fa fa-check"></i><b>9</b> Annotation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka"><i class="fa fa-check"></i><b>9.1</b> Prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-conda-environment-directory"><i class="fa fa-check"></i><b>9.1.1</b> Prokka: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-run"><i class="fa fa-check"></i><b>9.1.2</b> Prokka: Run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-output"><i class="fa fa-check"></i><b>9.1.3</b> Prokka: Output</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Final_report.html"><a href="10-Final_report.html"><i class="fa fa-check"></i><b>10</b> Final report</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report"><i class="fa fa-check"></i><b>10.1</b> Report</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>10.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="10.1.2" data-path="10-Final_report.html"><a href="10-Final_report.html#report-quast"><i class="fa fa-check"></i><b>10.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="10.1.3" data-path="10-Final_report.html"><a href="10-Final_report.html#report-busco"><i class="fa fa-check"></i><b>10.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="10.1.4" data-path="10-Final_report.html"><a href="10-Final_report.html#report-multiqc"><i class="fa fa-check"></i><b>10.1.4</b> Report: MultiQC</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Cleanup.html"><a href="11-Cleanup.html"><i class="fa fa-check"></i><b>11</b> Cleanup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Cleanup.html"><a href="11-Cleanup.html#quast-1"><i class="fa fa-check"></i><b>11.1</b> quast</a></li>
<li class="chapter" data-level="11.2" data-path="11-Cleanup.html"><a href="11-Cleanup.html#busco-1"><i class="fa fa-check"></i><b>11.2</b> busco</a></li>
<li class="chapter" data-level="11.3" data-path="11-Cleanup.html"><a href="11-Cleanup.html#checkm-1"><i class="fa fa-check"></i><b>11.3</b> checkm</a></li>
<li class="chapter" data-level="11.4" data-path="11-Cleanup.html"><a href="11-Cleanup.html#racon-1"><i class="fa fa-check"></i><b>11.4</b> racon</a></li>
<li class="chapter" data-level="11.5" data-path="11-Cleanup.html"><a href="11-Cleanup.html#all-prokkas-files-are-useful"><i class="fa fa-check"></i><b>11.5</b> All prokkas files are useful</a></li>
<li class="chapter" data-level="11.6" data-path="11-Cleanup.html"><a href="11-Cleanup.html#tar"><i class="fa fa-check"></i><b>11.6</b> tar</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="12-Appendix.html"><a href="12-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Circlator" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Circularisation</h1>
<p><img src="figures/circlator.png" width="18%" style="display: block; margin: auto;" /></p>
<p>Circularisation is useful for bacterial genomes which consist of circular chromosomes and plasmids.</p>
<div id="circlator" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Circlator</h2>
<p><img src="figures/chain_circle.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>Circlator</code>, the first circularisation tool created, attempts to identify circular sequences and produce a linearised version of them. The goal is to have each circular sequence (chromosome or plasmid) as one contig with no overlap between the start and end of the contig. This would lead to a contig where if you moved one position forward/right from the last base in the contig you would wrap back around to the first base of the contig.</p>
<p><img src="figures/circularisation_diagram.png" width="60%" style="display: block; margin: auto;" /></p>
<div id="circlator-conda-environment-directory" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Circlator: Conda environment &amp; directory</h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>circlator</code> conda environment for our <code>Circlator</code> circularisation.</p>
<p>Open a new terminal (right click on the main screen, choose <code>Applications</code> -&gt; <code>Shell</code> -&gt; <code>bash</code>) and run:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="08-Circlator.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="ex">chos</span> 8</span>
<span id="cb32-2"><a href="08-Circlator.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> usecirclator</span></code></pre></div>
<p>Move to the <code>standard_workflow</code> directory and make a new output directory for <code>Circlator</code>.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="08-Circlator.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Change directory</span></span>
<span id="cb33-2"><a href="08-Circlator.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow</span>
<span id="cb33-3"><a href="08-Circlator.html#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Make directory</span></span>
<span id="cb33-4"><a href="08-Circlator.html#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> circlator/redbean_racon_i2/</span></code></pre></div>
</div>
<div id="circlator-read-correction" class="section level3" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Circlator: Read correction</h3>
<p><img src="figures/correction.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>Circlator</code> requires corrected reads (reads with improved accuracy). There are various tools to correct reads (<a href="https://github.com/sanger-pathogens/circlator/wiki/Input-files-for-circlator" class="uri">https://github.com/sanger-pathogens/circlator/wiki/Input-files-for-circlator</a>). We will use the assembler <code>Canu</code> to correct our reads.</p>
<p>If you assemble your data with <code>Canu</code> you would already have the corrected reads. Since we have not we can correct them now using the correction part and skipping the assembly part.</p>
<p><strong>Note</strong>: The below command takes a long time. In the interest of time skip the command and use the lower down copy command to get the pre-computed corrected reads.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="08-Circlator.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="ex">canu</span> <span class="at">-correct</span> <span class="dt">\</span></span>
<span id="cb34-2"><a href="08-Circlator.html#cb34-2" aria-hidden="true" tabindex="-1"></a>-p ecoli <span class="dt">\</span></span>
<span id="cb34-3"><a href="08-Circlator.html#cb34-3" aria-hidden="true" tabindex="-1"></a>-d ecoli_reads <span class="dt">\</span></span>
<span id="cb34-4"><a href="08-Circlator.html#cb34-4" aria-hidden="true" tabindex="-1"></a>genomeSize=4.6m <span class="dt">\</span></span>
<span id="cb34-5"><a href="08-Circlator.html#cb34-5" aria-hidden="true" tabindex="-1"></a>-pacbio ecoli_reads/ecoli.fastq</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-p</code> : Prefix of output files.</li>
<li><code>-d</code> : Output directory.</li>
<li><code>genomeSize=</code> : Approximate genome size.</li>
<li><code>-pacbio</code> : Specify the pacbio reads to be used.
<ul>
<li>Alternative options are: <code>-nanopore</code> and <code>-pacbio-hifi</code>.</li>
</ul></li>
</ul>
<p>The above command would produce various files. However, we are only interested in the corrected reads. Copy the pre-computed corrected reads, these come in a <code>.fasta.gz</code> file, and gunzip the file for further use.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="08-Circlator.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Copy corrected read fasta.gz file</span></span>
<span id="cb35-2"><a href="08-Circlator.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> /pub39/tea/matthew/NEOF/bact_assembly/data/ecoli.correctedReads.fasta.gz ecoli_reads</span>
<span id="cb35-3"><a href="08-Circlator.html#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Gunzip file</span></span>
<span id="cb35-4"><a href="08-Circlator.html#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> ecoli_reads/ecoli.correctedReads.fasta.gz</span></code></pre></div>
</div>
<div id="circlator-circularisation" class="section level3" number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> Circlator: Circularisation</h3>
<p><img src="figures/circlularisation.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With the corrected reads we can now carry out the <code>Circlator</code> step. <code>Circlator</code> has many different commands for the various stages of circularisation:</p>
<ol start="0" style="list-style-type: decimal">
<li><strong>Check the input assembly</strong>: Formatting fasta headers.</li>
<li><strong>Map reads to assembly</strong>: Alignment of reads to the assmebly by <code>BWA mem</code>.</li>
<li><strong>Extract reads for reassembly</strong>: Extract the aligned reads, therefore removing reads that did not align to the assembly.</li>
<li><strong>Assembly of extract reads</strong>: Carried out with the <code>SPAdes</code> assembler.</li>
<li><strong>Merge and circularise</strong>: Merges overlapping contigs form the assembly. Once merging is complete circular sequences are identified. Circular sequences are arranged so the last base is followed by the first base.</li>
<li><strong>Clean contigs</strong>: Removal of small contigs (default = 2k) and contigs completely contained in another contig.</li>
<li><strong>Fix contig start positions</strong>: Attempts to make the start position to be at a dnaA gene (if found).
<ul>
<li>dnaA gene produces the “Chromosomal replication initiator protein DnaA” (<a href="https://www.uniprot.org/uniprot/P03004" class="uri">https://www.uniprot.org/uniprot/P03004</a>).</li>
</ul></li>
</ol>
<p>To run all 7 steps as a workflow We will use the <code>circlator all</code> command (<a href="https://github.com/sanger-pathogens/circlator/wiki/Task:-all" class="uri">https://github.com/sanger-pathogens/circlator/wiki/Task:-all</a>).</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="08-Circlator.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="ex">circlator</span> all <span class="dt">\</span></span>
<span id="cb36-2"><a href="08-Circlator.html#cb36-2" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i2.fasta <span class="dt">\</span></span>
<span id="cb36-3"><a href="08-Circlator.html#cb36-3" aria-hidden="true" tabindex="-1"></a>ecoli_reads/ecoli.correctedReads.fasta <span class="dt">\</span></span>
<span id="cb36-4"><a href="08-Circlator.html#cb36-4" aria-hidden="true" tabindex="-1"></a>circlator/redbean_racon_i2/ecoli</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li>The 1st flagless option is the input assembly to be circularised.</li>
<li>The 2nd flagless option is the corrected reads to be used for circularisation,</li>
<li>The 3rd flagless option is the output directory.</li>
</ul>
<p>You can list all the output noting the prefix number shows which step the files/directories were created at.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="08-Circlator.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> circlator/redbean_racon_i2/ecoli</span></code></pre></div>
<p>The final circularised genome is in the file <code>06.fixstart.fasta</code>.</p>
</div>
<div id="circlator-assessment-polishing" class="section level3" number="8.1.4">
<h3><span class="header-section-number">8.1.4</span> Circlator: Assessment &amp; Polishing</h3>
<p><img src="figures/assessment_n_polish.png" width="30%" style="display: block; margin: auto;" /></p>
<p>Now would be a good time to carry out a genome assessment to determine if circularisation was beneficial. Then it would be good to carry out further polishing.</p>
<p>You are more than welcome to do that now, if not we will go onto annotation.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="07-Polishing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="09-Annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
