<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Genome Assembly | Bacterial genome assembly and annotation</title>
  <meta name="description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Genome Assembly | Bacterial genome assembly and annotation" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Genome Assembly | Bacterial genome assembly and annotation" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2022-06-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="04-Overview.html"/>
<link rel="next" href="06-Genome_assessment.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation</a></li>

<li class="divider"></li>
<li class="part"><span><b>Start</b></span></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_assembly_and_annotation.html"><a href="01-Bacterial_genome_assembly_and_annotation.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Standard workflow</b></span></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Assembly.html"><a href="05-Assembly.html"><i class="fa fa-check"></i><b>5</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean"><i class="fa fa-check"></i><b>5.1</b> Redbean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbeanassemblyconda"><i class="fa fa-check"></i><b>5.1.1</b> Redbean: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-assemble-long-reads"><i class="fa fa-check"></i><b>5.1.2</b> Redbean: Assemble long reads</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-derive-consensus"><i class="fa fa-check"></i><b>5.1.3</b> Redbean: Derive consensus</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Assembly.html"><a href="05-Assembly.html#bandage-visualisation"><i class="fa fa-check"></i><b>5.2</b> Bandage Visualisation</a></li>
<li class="chapter" data-level="5.3" data-path="05-Assembly.html"><a href="05-Assembly.html#other-long-read-assemblers"><i class="fa fa-check"></i><b>5.3</b> Other long read assemblers</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html"><i class="fa fa-check"></i><b>6</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessmentconda"><i class="fa fa-check"></i><b>6.1</b> Assessment: Conda environnment &amp; directory</a></li>
<li class="chapter" data-level="6.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast"><i class="fa fa-check"></i><b>6.2</b> QUAST</a></li>
<li class="chapter" data-level="6.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>6.3</b> BUSCO</a></li>
<li class="chapter" data-level="6.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#checkm"><i class="fa fa-check"></i><b>6.4</b> CheckM</a></li>
<li class="chapter" data-level="6.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>6.5</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Polishing.html"><a href="07-Polishing.html"><i class="fa fa-check"></i><b>7</b> Polishing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon"><i class="fa fa-check"></i><b>7.1</b> Racon</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-conda-environment-directory"><i class="fa fa-check"></i><b>7.1.1</b> Racon: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-index-assembly"><i class="fa fa-check"></i><b>7.1.2</b> Racon: Index assembly</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-read-alignment-to-assembly"><i class="fa fa-check"></i><b>7.1.3</b> Racon: Read alignment to assembly</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing"><i class="fa fa-check"></i><b>7.1.4</b> Racon: Polishing</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing-report"><i class="fa fa-check"></i><b>7.1.5</b> Racon: Polishing report</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-second-iteration"><i class="fa fa-check"></i><b>7.1.6</b> Racon: Second iteration</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Polishing.html"><a href="07-Polishing.html#other-polishers"><i class="fa fa-check"></i><b>7.2</b> Other polishers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Circlator.html"><a href="08-Circlator.html"><i class="fa fa-check"></i><b>8</b> Circularisation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator"><i class="fa fa-check"></i><b>8.1</b> Circlator</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-conda-environment-directory"><i class="fa fa-check"></i><b>8.1.1</b> Circlator: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-read-correction"><i class="fa fa-check"></i><b>8.1.2</b> Circlator: Read correction</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-circularisation"><i class="fa fa-check"></i><b>8.1.3</b> Circlator: Circularisation</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-assessment-polishing"><i class="fa fa-check"></i><b>8.1.4</b> Circlator: Assessment &amp; Polishing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Annotation.html"><a href="09-Annotation.html"><i class="fa fa-check"></i><b>9</b> Annotation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka"><i class="fa fa-check"></i><b>9.1</b> Prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-conda-environment-directory"><i class="fa fa-check"></i><b>9.1.1</b> Prokka: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-run"><i class="fa fa-check"></i><b>9.1.2</b> Prokka: Run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-output"><i class="fa fa-check"></i><b>9.1.3</b> Prokka: Output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Annotation.html"><a href="09-Annotation.html#exploring-annotations"><i class="fa fa-check"></i><b>9.2</b> Exploring annotations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Final_report.html"><a href="10-Final_report.html"><i class="fa fa-check"></i><b>10</b> Final report</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report"><i class="fa fa-check"></i><b>10.1</b> Report</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>10.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="10.1.2" data-path="10-Final_report.html"><a href="10-Final_report.html#report-quast"><i class="fa fa-check"></i><b>10.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="10.1.3" data-path="10-Final_report.html"><a href="10-Final_report.html#report-busco"><i class="fa fa-check"></i><b>10.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="10.1.4" data-path="10-Final_report.html"><a href="10-Final_report.html#report-multiqc"><i class="fa fa-check"></i><b>10.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="10-Final_report.html"><a href="10-Final_report.html#final-recap"><i class="fa fa-check"></i><b>10.2</b> Final recap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Cleanup.html"><a href="11-Cleanup.html"><i class="fa fa-check"></i><b>11</b> Cleanup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Cleanup.html"><a href="11-Cleanup.html#deleting-unwanted-data"><i class="fa fa-check"></i><b>11.1</b> Deleting unwanted data</a></li>
<li class="chapter" data-level="11.2" data-path="11-Cleanup.html"><a href="11-Cleanup.html#compressing-the-final-directory"><i class="fa fa-check"></i><b>11.2</b> Compressing the final directory</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="09-Annotation.html"><a href="09-Annotation.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="redbeanassembly" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Genome Assembly<a href="05-Assembly.html#redbeanassembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/string.png" width="20%" style="display: block; margin: auto;" /></p>
<p>The first step of any genome assembly project is to carry out QC of the read data. The very next step is genome assembly. There are many different assemblers and in the past the choice of assembler and its parameters was the primary task to produce a good quality assembly after getting good quality reads.</p>
<p>The bioinformatician's choice of assembler and parameters is not as vital as it once was. This is due to great improvements to sequencing technologies and bioinformatics processes which include:</p>
<ul>
<li>Sequencing technologies are creating longer and higher quality reads.
<ul>
<li>The better the length and quality, the less work assemblers needs to do.</li>
<li>There may come a time when you can get one read that perfectly represents an entire genome/chromosome. Assembly would not be required at all.</li>
</ul></li>
<li>Genome assemblers are improving.
<ul>
<li>There is a large choice of good genome assemblers. It is less a question of which to use and more a question of which not to use.</li>
</ul></li>
<li>Genome assembly parameter choice is a smaller issue.
<ul>
<li>For older assemblers and older versions of assemblers it would take a lot of trial and error to try to find the best parameters.</li>
<li>Now like many bioinformatics tools, assemblers will attempt to use the best parameters for your data. Generally they are better and quicker than humans at this.</li>
</ul></li>
</ul>
<div id="redbean" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Redbean<a href="05-Assembly.html#redbean" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/redbean.png" width="15%" style="display: block; margin: auto;" /></p>
<p>For assembly of our PacBio reads we will be using <code>Redbean</code> (AKA <code>wtdbg2</code>). <code>Redbean</code> is a long read assembler with comparable performance to others. However, where it excels is its speed, it is much faster than other long read assemblers. This make it perfect for this workshop.</p>
<p><code>Redbean</code> can be used for genomes of all sizes. It will work for Viruses, Bacteria, Archaea, and Eukaryotes.</p>
<p><code>Redbean</code> works for PacBio RSII, PacBio Sequel, and Oxford nanopore data.</p>
<div id="redbeanassemblyconda" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Redbean: Conda environment &amp; directory<a href="05-Assembly.html#redbeanassemblyconda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>bacterial_assembly</code> <code>conda</code> environment for our <code>Redbean</code> assembly. This will be our main conda environment for our standard workflow.</p>
<p>Activate the environment:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="05-Assembly.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> usebacterialassembly</span></code></pre></div>
<p>Before we start we will move into our analysis directory (created in <a href="03-Start.html#setup">setup</a>) and create a directory for the assembly we will produce.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="05-Assembly.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Change directory to analysis directory</span></span>
<span id="cb5-2"><a href="05-Assembly.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow</span>
<span id="cb5-3"><a href="05-Assembly.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Create directory for redbean output</span></span>
<span id="cb5-4"><a href="05-Assembly.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> redbean_assembly</span></code></pre></div>
</div>
<div id="redbean-assemble-long-reads" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Redbean: Assemble long reads<a href="05-Assembly.html#redbean-assemble-long-reads" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/assembly_simple.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The first step of <code>Redbean</code> is to carry out the assembly. This is carried out with the <code>wtdgb2</code> command.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="05-Assembly.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">wtdbg2</span> <span class="at">-x</span> rs <span class="at">-g</span> 4.6m <span class="at">-i</span> ecoli_reads/ecoli.fastq <span class="at">-o</span> redbean_assembly/ecoli <span class="at">-t</span> 8</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-x</code> : This indicates the sequencing technology of the input reads.
<ul>
<li><code>rs</code> = PacBio RSII</li>
<li><code>sq</code> = PacBio Sequel</li>
<li><code>ccs</code> = PacBio CCS reads</li>
<li><code>ont</code> = Oxford Nanopore</li>
</ul></li>
<li><code>-g</code> : This indicates the estimated size of the genome
<ul>
<li>This does not need to be super exact with and you can round up to two significant figures. Examples are below.
<ul>
<li><em>E.coli</em> | genome size = 4,639,221 bp =&gt; <code>-g 4.6m</code> (m = megabases)</li>
<li><em>Phi-X174</em> | genome size = 5386 bp =&gt; <code>-g 5.4k</code> (k = kilobase)</li>
<li><em>H.Sapiens</em> | genome size = 3.2Gb =&gt; <code>-g 3g</code> (g = gigabase)</li>
</ul></li>
</ul></li>
<li><code>-i</code> : The input reads.</li>
<li><code>-o</code> : The output prefix.
<ul>
<li>With this set to <code>redbean_assembly/ecoli</code> all the output files will be in the directory <code>redbean_assembly</code> with the prefix <code>ecoli</code>.</li>
</ul></li>
<li><code>-t</code> : Number of threads to be used.</li>
</ul>
<p>The will create a bunch of output files. Most of these we will ignore with only 4 being used below.</p>
</div>
<div id="redbean-derive-consensus" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Redbean: Derive consensus<a href="05-Assembly.html#redbean-derive-consensus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/consensus.png" width="10%" style="display: block; margin: auto;" /></p>
<p>If you checked the output directory you may have noticed there was no fasta file. Therefore the next step is to create a fasta file that contains the consensus assembly with <code>wtpoa-cns</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="05-Assembly.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">wtpoa-cns</span> <span class="at">-t</span> 8 <span class="at">-i</span> redbean_assembly/ecoli.ctg.lay.gz <span class="at">-fo</span> redbean_assembly/ecoli.ctg.fa</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-i</code> : The input <code>ctg.lay.gz</code> file produced by <code>wtdbg2</code></li>
<li><code>-fo</code> : The output file path for the consensus assembly in fasta format.</li>
<li><code>-t</code> : Number of threads to be used.</li>
</ul>
<p>The consensus is produced from the assembly graph (more on the graph below). Assembly is not always straight forward and when producing a graph, an assembler may derive multiple paths/branches through the assembly.</p>
<p>Two examples of how this may occur are:</p>
<ul>
<li><strong>Repeating regions</strong>: An assembler may find it hard to determine how many times a repeat occurs within the repeat region. Maybe it is 10 or 11 times.</li>
<li><strong>Identical homopolymers</strong>: It is possible that long homopolymers of As (or one of the other bases) occurs multiple times in the genome. It may then be difficult to know what is at either end of each unique homopolymer location.
<ul>
<li>E.g. There are 2 homopolymer sequences (H1 &amp; H2) with 4 ends (E1-4).
<ul>
<li>Is H1 flanked by E1 and E2 and therefore H2 is flanked by E3 and E4?</li>
<li>Or is H1 flanked by E1 and E3 and therefore H2 is flanked by E2 and E4?</li>
<li>Etc.</li>
</ul></li>
</ul></li>
</ul>
<p>The consensus can then be thought of as the one best path through the graph to produce the most likely genome assembly.</p>
</div>
</div>
<div id="bandage-visualisation" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Bandage Visualisation<a href="05-Assembly.html#bandage-visualisation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/bandage.png" width="15%" style="display: block; margin: auto;" /></p>
<p>This part is optional and generally something I never do in real analyses. However, it is interesting and will help your understanding of assembly graphs.</p>
<p><code>Redbean</code> carries out 3 iterations of graph construction, each getting better. These three graph iterations are represented by the <code>.dot.gz</code> files. We can visualise these graphs with the tool <code>Bandage</code>.</p>
<p>Before visualisation we need to convert the files to <code>.gfa</code> (link on file specs in <a href="09-Annotation.html#manuals">appendix</a>) files which are compatible with <code>Bandage</code>.</p>
<p>First we need to <code>gunzip</code> the files.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="05-Assembly.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> redbean_assembly/ecoli.1.dot.gz</span>
<span id="cb8-2"><a href="05-Assembly.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> redbean_assembly/ecoli.2.dot.gz</span>
<span id="cb8-3"><a href="05-Assembly.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> redbean_assembly/ecoli.3.dot.gz</span></code></pre></div>
<p>Next we convert the <code>.dot</code> files into <code>.gfa</code> files.</p>
<p>The script <code>wtdbg2-dot2gfa.pl</code> is found in the scripts directory of <code>wtdbg2</code> which can be found at: <a href="https://github.com/ruanjue/wtdbg2/tree/master/scripts" class="uri">https://github.com/ruanjue/wtdbg2/tree/master/scripts</a>. Unfortunately this is not installed when you install <code>wtdbg2</code> via <code>Conda</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="05-Assembly.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">wtdbg-dot2gfa.pl</span> redbean_assembly/ecoli.1.dot <span class="op">&gt;</span> redbean_assembly/ecoli.1.dot.gfa</span>
<span id="cb9-2"><a href="05-Assembly.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">wtdbg-dot2gfa.pl</span> redbean_assembly/ecoli.2.dot <span class="op">&gt;</span> redbean_assembly/ecoli.2.dot.gfa</span>
<span id="cb9-3"><a href="05-Assembly.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">wtdbg-dot2gfa.pl</span> redbean_assembly/ecoli.3.dot <span class="op">&gt;</span> redbean_assembly/ecoli.3.dot.gfa</span></code></pre></div>
<p>Now we can open up the <code>Bandage</code> GUI.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="05-Assembly.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Bandage</span></span></code></pre></div>
<p>Using this GUI we will open the <code>.gfa</code> files one at a time. Start with the <code>ecoli.1.dot.gfa</code> file (the 1st graph), then the <code>ecoli.2.dot.gfa</code> file (2nd), followed by the <code>ecoli.3.dot.gfa</code> file (3rd, last, and best).</p>
<p>To open a graph carry out the following</p>
<ul>
<li>On the tool bar click <code>File</code> -&gt; <code>Load graph</code></li>
<li>In the pop up file explorer navigate to <code>~/bacterial_assembly/standard_workflow/redbean_assembly/</code> and double click on the <code>.gfa</code> file you would like to view.</li>
<li>In the <code>Graph drawing</code> section on the right side bar:
<ul>
<li>Ensure <code>Scope:</code> is set to <code>Entire graph</code></li>
<li>Ensure <code>Style</code> is set to <code>Single</code></li>
<li>Click the <code>Draw graph</code> button</li>
</ul></li>
</ul>
<p><strong>Tips</strong>:</p>
<ul>
<li>The graph for <code>ecoli.1.dot.gfa</code> will take quite a while to load.</li>
<li>The graphs for <code>ecoli.2.dot.gfa</code> and <code>ecoli.3.dot.gfa</code> are very large so you may need to change the <code>zoom</code> to 1%.</li>
</ul>
<p>As you go from 1 to 3 you will see two main changes:</p>
<ul>
<li><strong>Fewer contigs</strong>: This shows the assembler is connecting parts of the assembly to give a more contiguous assembly.</li>
<li><strong>Fewer branches</strong>: In the 1st and 2nd graphs you will see branches, parts of the graph that branch off and possibly reconnect again. This represents multiple paths through the assembly that are possible. Through the <code>Redbean</code> iterations the assembler chooses the best routes, removing the other routes. You will need to zoom in to properly see these branches.</li>
</ul>
</div>
<div id="other-long-read-assemblers" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Other long read assemblers<a href="05-Assembly.html#other-long-read-assemblers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/canoe.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>Redbean</code> will not always be the optimal assembler. However, even if you plan to use another assembler <code>Redbean</code> is very good as a first pass tool due to its speed. It can be very useful to run a <code>Redbean</code> assembly and carry out genome assessment (shown in next chapter) to see if there are any issues with the reads that was not caught in the read QC. Various issues and solutions are covered in the supplemental bookdown.</p>
<p>Other long read assemblers to explore for yourself include (links in <a href="09-Annotation.html#manuals">appendix</a>):</p>
<ul>
<li><code>Canu</code></li>
<li><code>HGAP</code></li>
<li><code>Flye</code></li>
<li><code>Raven</code></li>
<li><code>NECAT</code></li>
<li><code>Shasta</code></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="04-Overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="06-Genome_assessment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
