<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Genome assembly assessment | Bacterial genome assembly and annotation</title>
  <meta name="description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Genome assembly assessment | Bacterial genome assembly and annotation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Genome assembly assessment | Bacterial genome assembly and annotation" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2021-08-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="05-Assembly.html"/>
<link rel="next" href="07-Polishing.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation</a></li>

<li class="divider"></li>
<li class="part"><span><b>Start</b></span></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_assembly_and_annotation.html"><a href="01-Bacterial_genome_assembly_and_annotation.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Standard workflow</b></span></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Assembly.html"><a href="05-Assembly.html"><i class="fa fa-check"></i><b>5</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean"><i class="fa fa-check"></i><b>5.1</b> Redbean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbeanassemblyconda"><i class="fa fa-check"></i><b>5.1.1</b> Redbean: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-assemble-long-reads"><i class="fa fa-check"></i><b>5.1.2</b> Redbean: Assemble long reads</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-derive-consensus"><i class="fa fa-check"></i><b>5.1.3</b> Redbean: Derive consensus</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Assembly.html"><a href="05-Assembly.html#bandage-visualisation"><i class="fa fa-check"></i><b>5.2</b> Bandage Visualisation</a></li>
<li class="chapter" data-level="5.3" data-path="05-Assembly.html"><a href="05-Assembly.html#other-long-read-assemblers"><i class="fa fa-check"></i><b>5.3</b> Other long read assemblers</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html"><i class="fa fa-check"></i><b>6</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessmentconda"><i class="fa fa-check"></i><b>6.1</b> Assessment: Conda environnment &amp; directory</a></li>
<li class="chapter" data-level="6.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast"><i class="fa fa-check"></i><b>6.2</b> QUAST</a></li>
<li class="chapter" data-level="6.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>6.3</b> BUSCO</a></li>
<li class="chapter" data-level="6.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#checkm"><i class="fa fa-check"></i><b>6.4</b> CheckM</a></li>
<li class="chapter" data-level="6.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>6.5</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Polishing.html"><a href="07-Polishing.html"><i class="fa fa-check"></i><b>7</b> Polishing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon"><i class="fa fa-check"></i><b>7.1</b> Racon</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-conda-environment-directory"><i class="fa fa-check"></i><b>7.1.1</b> Racon: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-index-assembly"><i class="fa fa-check"></i><b>7.1.2</b> Racon: Index assembly</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-read-alignment-to-assembly"><i class="fa fa-check"></i><b>7.1.3</b> Racon: Read alignment to assembly</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing"><i class="fa fa-check"></i><b>7.1.4</b> Racon: Polishing</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing-report"><i class="fa fa-check"></i><b>7.1.5</b> Racon: Polishing report</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-second-iteration"><i class="fa fa-check"></i><b>7.1.6</b> Racon: Second iteration</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Polishing.html"><a href="07-Polishing.html#other-polishers"><i class="fa fa-check"></i><b>7.2</b> Other polishers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Circlator.html"><a href="08-Circlator.html"><i class="fa fa-check"></i><b>8</b> Circularisation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator"><i class="fa fa-check"></i><b>8.1</b> Circlator</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-conda-environment-directory"><i class="fa fa-check"></i><b>8.1.1</b> Circlator: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-read-correction"><i class="fa fa-check"></i><b>8.1.2</b> Circlator: Read correction</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-circularisation"><i class="fa fa-check"></i><b>8.1.3</b> Circlator: Circularisation</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-assessment-polishing"><i class="fa fa-check"></i><b>8.1.4</b> Circlator: Assessment &amp; Polishing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Annotation.html"><a href="09-Annotation.html"><i class="fa fa-check"></i><b>9</b> Annotation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka"><i class="fa fa-check"></i><b>9.1</b> Prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-conda-environment-directory"><i class="fa fa-check"></i><b>9.1.1</b> Prokka: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-run"><i class="fa fa-check"></i><b>9.1.2</b> Prokka: Run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-output"><i class="fa fa-check"></i><b>9.1.3</b> Prokka: Output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Annotation.html"><a href="09-Annotation.html#exploring-annotations"><i class="fa fa-check"></i><b>9.2</b> Exploring annotations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Final_report.html"><a href="10-Final_report.html"><i class="fa fa-check"></i><b>10</b> Final report</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report"><i class="fa fa-check"></i><b>10.1</b> Report</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>10.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="10.1.2" data-path="10-Final_report.html"><a href="10-Final_report.html#report-quast"><i class="fa fa-check"></i><b>10.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="10.1.3" data-path="10-Final_report.html"><a href="10-Final_report.html#report-busco"><i class="fa fa-check"></i><b>10.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="10.1.4" data-path="10-Final_report.html"><a href="10-Final_report.html#report-multiqc"><i class="fa fa-check"></i><b>10.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="10-Final_report.html"><a href="10-Final_report.html#final-recap"><i class="fa fa-check"></i><b>10.2</b> Final recap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Cleanup.html"><a href="11-Cleanup.html"><i class="fa fa-check"></i><b>11</b> Cleanup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Cleanup.html"><a href="11-Cleanup.html#deleting-unwanted-data"><i class="fa fa-check"></i><b>11.1</b> Deleting unwanted data</a></li>
<li class="chapter" data-level="11.2" data-path="11-Cleanup.html"><a href="11-Cleanup.html#compressing-the-final-directory"><i class="fa fa-check"></i><b>11.2</b> Compressing the final directory</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="09-Annotation.html"><a href="09-Annotation.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomeassemblyassessment" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Genome assembly assessment</h1>
<p><img src="figures/inspect.png" width="20%" style="display: block; margin: auto;" /></p>
<p>After carrying out an assembly it is always advised to assess its quality. We will carry this out with three tools:</p>
<ul>
<li><code>QUAST</code> - Gives assembly stats related to contiguity.</li>
<li><code>BUSCO</code> - Estimates completeness of a genome.</li>
<li><code>CheckM</code> - Estimates completeness and contamination of a genome.</li>
</ul>
<div id="assessmentconda" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Assessment: Conda environnment &amp; directory</h2>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Open a new terminal (right click on the main screen, choose <code>Applications</code> -&gt; <code>Shell</code> -&gt; <code>bash</code>).</p>
<p>We will use the <code>bacterial_genome_assessment</code> <code>conda</code> environment for our genome assessment tools.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="06-Genome_assessment.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">chos</span> 8</span>
<span id="cb11-2"><a href="06-Genome_assessment.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> usebacterialgenomeassessment</span></code></pre></div>
<p>Ensure you are in the the <code>standard_workflow</code> directory for this new terminal and environment.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="06-Genome_assessment.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow</span></code></pre></div>
</div>
<div id="quast" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> QUAST</h2>
<p><img src="figures/measuring_tape.png" width="10%" style="display: block; margin: auto;" /></p>
<p><code>QUAST</code> is the most used tool to assess genomes. It gives various statistics related to the contiguity of the assembly. Unless stated these statistics are based on contigs that are &gt;=500bp.</p>
<p>Create an output directory and run <code>QUAST</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="06-Genome_assessment.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make output directory</span></span>
<span id="cb13-2"><a href="06-Genome_assessment.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> quast/redbean</span>
<span id="cb13-3"><a href="06-Genome_assessment.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Run QUAST</span></span>
<span id="cb13-4"><a href="06-Genome_assessment.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="ex">quast</span> <span class="at">-o</span> quast/redbean redbean_assembly/ecoli.ctg.fa</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-o</code> : The output directory.</li>
<li>The input file/s are at the end of the command without a flag (e.g. <code>-o</code>).
<ul>
<li><strong>Note</strong> : <code>QUAST</code> can be run on multiple assemblies giving a report which contains info on all the input assemblies.</li>
</ul></li>
</ul>
<p><code>QUAST</code> produces reports in various formats. We will look at the html report.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="06-Genome_assessment.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">firefox</span> quast/redbean/report.html</span></code></pre></div>
<p>Inspect the table and interact with the interactive plots.</p>
<p>Statistics such as length, # contigs (number of contigs), and GC (%) are self explanatory. Two definitions that you may not be aware of are N50/N75 and L50/L75.</p>
<p>To calculate N50 and L50:</p>
<ol style="list-style-type: decimal">
<li>Order the contigs from largest to smallest.</li>
<li>Find the point that is 50% the length of the assembly starting from the start of the largest contig going toward the smallest contig.</li>
<li>Determine which contig this point belongs to.</li>
<li>The N50 is equal to the length of the contig.</li>
<li>The L50 is equal to the number of contigs that is &gt;= in length to the N50.</li>
</ol>
<p>To find the N75 &amp; L75 change the 50% to 75% in point 2. You can use any number between 1-100 for the Nx or Lx.</p>
<p>Additionally, below is a visual demonstration of Nx and Lx.</p>
<p><img src="figures/n50_n90.png" width="60%" style="display: block; margin: auto;" /></p>
<p><strong>Questions</strong></p>
<ul>
<li>How many total contigs are there?</li>
<li>Approximately how much bigger is the total assembly compared to the largest contigs (to the nearest 1k)?</li>
<li>Does the GC% match what is know about <em>E.coli</em>? (You may need to google)</li>
<li>Why is the N50 and N75 the same?</li>
</ul>
<p>Overall these contiguity stats are good. It is not a one contig assembly, which is what we would aim for a single chromosome genome, but it is close. It mainly consists of one large contig with 3 smaller contigs.</p>
<p>Some things you will want to look out for in future projects:</p>
<ul>
<li>High number of contigs, with small N50: This may indicate the genome assembly is very uncontiguous and so the assembler had issues with assembly.
<ul>
<li>It is possible the genome is a particularly difficult genome to resolve.</li>
<li>The input data could be poor. Examples include:
<ul>
<li>Low quality.</li>
<li>Low coverage.</li>
<li>Average coverage could be good but it is possible areas of the genome have low coverage so these areas could not be assembled well.</li>
</ul></li>
</ul></li>
<li>Length is not correct: Could be caused by several reasons:
<ul>
<li>Parts of the genome are duplicated in the assembly causing the assembly to be too long. This can be caused by parts where a consensus could not be reached effectively.</li>
<li>A short assembly can indicate that parts of the genome are missing.</li>
<li>A drastically increased length can indicate the presence of contamination from another organism. The GC (%) can give a clue if this may be the case.</li>
</ul></li>
</ul>
<p>Further assessment can elucidate if there are any issues and what the issue may be.</p>
</div>
<div id="busco" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> BUSCO</h2>
<p><img src="figures/universe.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>BUSCO</code> stands for Benchmarking Universal Single-Copy Orthologs. The tool is used to estimate the completeness of genome assemblies by determining how many BUSCOs are present.</p>
<p>Prior to running <code>BUSCO</code> we will create an output directory for the <code>BUSCO</code> analysis.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="06-Genome_assessment.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> busco/redbean</span></code></pre></div>
<p>BUSCOs are genes that are:</p>
<ul>
<li><strong>Universal</strong>: The genes are present in all organisms of a specific lineage.</li>
<li><strong>Single-Copy</strong>: The genes are only found as one copy.</li>
</ul>
<p><code>BUSCO</code> has many different lineage datasets. These are datasets of the different groups of BUSCOs found in various different lineages. Carry out the below command to see all these different datasets.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="06-Genome_assessment.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="at">--list-datasets</span> <span class="kw">|</span> <span class="fu">less</span> <span class="at">-S</span></span></code></pre></div>
<p>You will notice that the lineage datasets are at various taxonomic levels going from kingdom (bacteria, archaea, &amp; eukaryota) down to order (rhizobiales, burkholderiales etc.). It is always good to be as specific as you can to your organisms lineage. A more specific lineage will have more BUSCOs that a more generic one.</p>
<p><strong>Note</strong>: All datasets end with <code>_odb10</code> which stands for OrthoDB version 10 (<a href="https://www.orthodb.org/" class="uri">https://www.orthodb.org/</a>).</p>
<p>Looking through the choices (and using NCBI taxonomy) what lineage dataset would you use for the following genera?</p>
<ul>
<li><strong>Psychrobacter</strong> (Taxonomy ID: 497)</li>
<li><strong>Iamia</strong> (Taxonomy ID: 467975)</li>
<li><strong>Phycisphaera</strong> (Taxonomy ID: 666508)</li>
<li><strong>Atribacter</strong> (Taxonomy ID: 2847777)</li>
</ul>
<p>For our <code>E.coli</code> we will choose the <code>enterobacterales_odb10</code> as our lineage dataset. With that we can run <code>BUSCO</code> on our genome assembly.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="06-Genome_assessment.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="dt">\</span></span>
<span id="cb17-2"><a href="06-Genome_assessment.html#cb17-2" aria-hidden="true" tabindex="-1"></a>-i redbean_assembly/ecoli.ctg.fa <span class="dt">\</span></span>
<span id="cb17-3"><a href="06-Genome_assessment.html#cb17-3" aria-hidden="true" tabindex="-1"></a>-l enterobacterales_odb10 <span class="dt">\</span></span>
<span id="cb17-4"><a href="06-Genome_assessment.html#cb17-4" aria-hidden="true" tabindex="-1"></a>-m geno <span class="dt">\</span></span>
<span id="cb17-5"><a href="06-Genome_assessment.html#cb17-5" aria-hidden="true" tabindex="-1"></a>-o ecoli <span class="dt">\</span></span>
<span id="cb17-6"><a href="06-Genome_assessment.html#cb17-6" aria-hidden="true" tabindex="-1"></a>--out_path busco/redbean</span></code></pre></div>
<p><strong>Parameter choice</strong></p>
<ul>
<li><code>-i</code> : The input genome assembly in fasat format.</li>
<li><code>-l</code> : The lineage dataset to be used. This has to end with the <code>_odb10</code> part. This will download the dataset (if it does not exist) into a directory called <code>busco_downloads</code> in the current directory. The directory will be created if it doesn’t exist.</li>
<li><code>-m</code> : The assessment mode.
<ul>
<li><code>geno</code> = genome assemblies (DNA)</li>
<li><code>tran</code> = transcriptome assemblies (DNA)</li>
<li><code>prot</code> = annotated gene sets (protein)</li>
</ul></li>
<li><code>-o</code> : The run name. Output folders and files will be labelled with this name.</li>
<li><code>--out_path</code> : Path where the output directory will be created.</li>
</ul>
<p>There are many output files in the directory <code>busco/redbean/ecoli</code>. A full list of the output can be found in the following link: <a href="https://busco.ezlab.org/busco_userguide.html#outputs" class="uri">https://busco.ezlab.org/busco_userguide.html#outputs</a></p>
<p>I ignore most of the output most of the time, instead only looking at the short summary file.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="06-Genome_assessment.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> busco/redbean/ecoli/short_summary.specific.enterobacterales_obd10.ecoli.txt</span></code></pre></div>
<p>This provides us with 6 values. These values are presented in terms of percentage (along the top line of the Results) and total numbers (the next 6 lines). The values represent:</p>
<ul>
<li><strong>Complete BUSCOs</strong> : The number of complete BUSCOs discovered. The closer this value is to 100% the better. This number is the sum of the next two values.</li>
<li><strong>Complete and single-copy BUSCOs</strong>: The number of BUSCOs found as complete and single-copy. The closer this number is to 100% the better. A good assembly will generally be &gt;90%.</li>
<li><strong>Complete and duplicated BUSCOs</strong>: Complete BUSCOs that have been found two or more times within the assembly. This indicates there are duplicated areas of the genome. Preferably we would like this number to be low.</li>
<li><strong>Fragmented BUSCOs</strong>: This could indicate that genes that only partially present within the genome. This could be due to misassmebly.</li>
<li><strong>Missing BUSCOs</strong>: These are BUSCOs that could not be found in the genome assembly. This could indicate parts of the genome are missing in the assembly.</li>
<li><strong>Total BUSCO groups searched</strong>: This represents the total number of BUSCOs within lineage dataset used for analysis.</li>
</ul>
<p>For more info on interpreting the results please see the following link: <a href="https://busco.ezlab.org/busco_userguide.html#interpreting-the-results" class="uri">https://busco.ezlab.org/busco_userguide.html#interpreting-the-results</a></p>
<p><strong>Questions</strong></p>
<ul>
<li>Ignoring total BUSCOs, what is the highest value? Do you want this value to be the highest?</li>
<li>Are these values good? What may the stats indicate?</li>
</ul>
<p>The <code>BUSCO</code> stats are ok. There are a bit more fragmented and missing BUSCOs than we would like. We’ll try to improve this with polishing. We could also try to get some better data (maybe some Illumina data) or try a different assembler. However, as this is a workshop we will continue on.</p>
</div>
<div id="checkm" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> CheckM</h2>
<p><img src="figures/CheckM.png" width="30%" style="display: block; margin: auto;" /></p>
<p><code>CheckM</code> can also be used to estimate the completeness of a bacterial genome, however I prefer <code>BUSCO</code> for this. Instead, we will use <code>CheckM</code> to estimate contamination within our assembly.</p>
<p>First create an output directory.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="06-Genome_assessment.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> checkm/redbean/ecoli</span></code></pre></div>
<p><code>CheckM</code> has many different commands. We will use its typical workflow which is conveniently wrapped into one command.</p>
<p><strong>Note</strong>: This command can take &gt;10 minutes.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="06-Genome_assessment.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">checkm</span> lineage_wf <span class="dt">\</span></span>
<span id="cb20-2"><a href="06-Genome_assessment.html#cb20-2" aria-hidden="true" tabindex="-1"></a>--tab_table <span class="dt">\</span></span>
<span id="cb20-3"><a href="06-Genome_assessment.html#cb20-3" aria-hidden="true" tabindex="-1"></a>-f checkm/redbean/ecoli/ecoli_checkm.tsv <span class="dt">\</span></span>
<span id="cb20-4"><a href="06-Genome_assessment.html#cb20-4" aria-hidden="true" tabindex="-1"></a>-t 8 <span class="dt">\</span></span>
<span id="cb20-5"><a href="06-Genome_assessment.html#cb20-5" aria-hidden="true" tabindex="-1"></a>-x ctg.fa redbean_assembly/ <span class="dt">\</span></span>
<span id="cb20-6"><a href="06-Genome_assessment.html#cb20-6" aria-hidden="true" tabindex="-1"></a>checkm/redbean/ecoli</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>--tab_table</code> : Creates a <code>.tsv</code> file (tab seperated value) that acts a summary of results. This is the one file we will inspect.</li>
<li><code>-f</code> : Path for output summary <code>.tsv</code> file.</li>
<li><code>-t</code> : Number of threads to be used.</li>
<li><code>-x</code> : Suffix of genome assembly files to be analysed. The command will carry out analysis on all the files with this suffix in the <strong>bin directory</strong>.</li>
<li>The second last option (which is flagless) is the <strong>bin directory</strong>. The command will carry out the workflow on each file with the suffix as stated in the <code>-x</code> parameter.</li>
<li>The last option (which is flagless) is the output directory.</li>
</ul>
<p>Ignore all the other files and view the summary files.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="06-Genome_assessment.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> checkm/redbean/ecoli/ecoli_checkm.tsv</span></code></pre></div>
<p>There are quite a few statistics but we are only interested in the <code>Contamination</code> column (you may need to use the right key to move to it). This value is a percentage. Is it a large value?</p>
<p>Thankfully not, it is less than 1%. If this was larger than 1% we may need to worry. If that was the case we could try to determine if it is real contamination and remove it if so. Please check out Blobtools2 if you ar einterested in contamination removal (link in <a href="09-Annotation.html#manuals">Appendix</a>)</p>
<p>If you are interested in the other values please check: <a href="https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics" class="uri">https://github.com/Ecogenomics/CheckM/wiki/Reported-Statistics</a></p>
</div>
<div id="assessment-recap" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Assessment recap</h2>
<p><img src="figures/recap.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Our assembly looks good. Good contiguity figures from <code>QUAST</code>, and <code>CheckM</code> shows low signs, if any, contamination. The <code>BUSCO</code> values are not brilliant with some fragmented and missing BUSCOs.</p>
<p>In real analysis I would try another assembler to see if that would improve the BUSCO situation. For the purpose of time we will continue and I will let you explore other assemblers in your own time.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="05-Assembly.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-Polishing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
