<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Polishing | Bacterial genome assembly and annotation</title>
  <meta name="description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Polishing | Bacterial genome assembly and annotation" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Polishing | Bacterial genome assembly and annotation" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2022-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="06-Genome_assessment.html"/>
<link rel="next" href="08-Circlator.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation</a></li>

<li class="divider"></li>
<li class="part"><span><b>Start</b></span></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_assembly_and_annotation.html"><a href="01-Bacterial_genome_assembly_and_annotation.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Standard workflow</b></span></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Assembly.html"><a href="05-Assembly.html"><i class="fa fa-check"></i><b>5</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean"><i class="fa fa-check"></i><b>5.1</b> Redbean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbeanassemblyconda"><i class="fa fa-check"></i><b>5.1.1</b> Redbean: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-assemble-long-reads"><i class="fa fa-check"></i><b>5.1.2</b> Redbean: Assemble long reads</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-derive-consensus"><i class="fa fa-check"></i><b>5.1.3</b> Redbean: Derive consensus</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Assembly.html"><a href="05-Assembly.html#bandage-visualisation"><i class="fa fa-check"></i><b>5.2</b> Bandage Visualisation</a></li>
<li class="chapter" data-level="5.3" data-path="05-Assembly.html"><a href="05-Assembly.html#other-long-read-assemblers"><i class="fa fa-check"></i><b>5.3</b> Other long read assemblers</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html"><i class="fa fa-check"></i><b>6</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessmentconda"><i class="fa fa-check"></i><b>6.1</b> Assessment: Conda environnment &amp; directory</a></li>
<li class="chapter" data-level="6.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast"><i class="fa fa-check"></i><b>6.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast-questions"><i class="fa fa-check"></i><b>6.2.1</b> QUAST Questions</a></li>
<li class="chapter" data-level="6.2.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast-summary"><i class="fa fa-check"></i><b>6.2.2</b> QUAST summary</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>6.3</b> BUSCO</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>6.3.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>6.3.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>6.3.3</b> BUSCO output</a></li>
<li class="chapter" data-level="6.3.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>6.3.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="6.3.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>6.3.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#checkm"><i class="fa fa-check"></i><b>6.4</b> CheckM</a></li>
<li class="chapter" data-level="6.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>6.5</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Polishing.html"><a href="07-Polishing.html"><i class="fa fa-check"></i><b>7</b> Polishing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon"><i class="fa fa-check"></i><b>7.1</b> Racon</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-conda-environment-directory"><i class="fa fa-check"></i><b>7.1.1</b> Racon: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-index-assembly"><i class="fa fa-check"></i><b>7.1.2</b> Racon: Index assembly</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-read-alignment-to-assembly"><i class="fa fa-check"></i><b>7.1.3</b> Racon: Read alignment to assembly</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing"><i class="fa fa-check"></i><b>7.1.4</b> Racon: Polishing</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing-report"><i class="fa fa-check"></i><b>7.1.5</b> Racon: Polishing report</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-second-iteration"><i class="fa fa-check"></i><b>7.1.6</b> Racon: Second iteration</a></li>
<li class="chapter" data-level="7.1.7" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-questions-summary"><i class="fa fa-check"></i><b>7.1.7</b> Racon; Questions &amp; Summary</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Polishing.html"><a href="07-Polishing.html#other-polishers"><i class="fa fa-check"></i><b>7.2</b> Other polishers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Circlator.html"><a href="08-Circlator.html"><i class="fa fa-check"></i><b>8</b> Circularisation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator"><i class="fa fa-check"></i><b>8.1</b> Circlator</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-conda-environment-directory"><i class="fa fa-check"></i><b>8.1.1</b> Circlator: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-read-correction"><i class="fa fa-check"></i><b>8.1.2</b> Circlator: Read correction</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-circularisation"><i class="fa fa-check"></i><b>8.1.3</b> Circlator: Circularisation</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-assessment-polishing"><i class="fa fa-check"></i><b>8.1.4</b> Circlator: Assessment &amp; Polishing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Annotation.html"><a href="09-Annotation.html"><i class="fa fa-check"></i><b>9</b> Annotation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka"><i class="fa fa-check"></i><b>9.1</b> Prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-conda-environment-directory"><i class="fa fa-check"></i><b>9.1.1</b> Prokka: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-run"><i class="fa fa-check"></i><b>9.1.2</b> Prokka: Run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-output"><i class="fa fa-check"></i><b>9.1.3</b> Prokka: Output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Annotation.html"><a href="09-Annotation.html#exploring-annotations"><i class="fa fa-check"></i><b>9.2</b> Exploring annotations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Final_report.html"><a href="10-Final_report.html"><i class="fa fa-check"></i><b>10</b> Final report</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report"><i class="fa fa-check"></i><b>10.1</b> Report</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>10.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="10.1.2" data-path="10-Final_report.html"><a href="10-Final_report.html#report-quast"><i class="fa fa-check"></i><b>10.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="10.1.3" data-path="10-Final_report.html"><a href="10-Final_report.html#report-busco"><i class="fa fa-check"></i><b>10.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="10.1.4" data-path="10-Final_report.html"><a href="10-Final_report.html#report-multiqc"><i class="fa fa-check"></i><b>10.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="10-Final_report.html"><a href="10-Final_report.html#final-recap"><i class="fa fa-check"></i><b>10.2</b> Final recap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Cleanup.html"><a href="11-Cleanup.html"><i class="fa fa-check"></i><b>11</b> Cleanup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Cleanup.html"><a href="11-Cleanup.html#deleting-unwanted-data"><i class="fa fa-check"></i><b>11.1</b> Deleting unwanted data</a></li>
<li class="chapter" data-level="11.2" data-path="11-Cleanup.html"><a href="11-Cleanup.html#compressing-the-final-directory"><i class="fa fa-check"></i><b>11.2</b> Compressing the final directory</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="09-Annotation.html"><a href="09-Annotation.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="polishing" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Polishing<a href="07-Polishing.html#polishing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/polishing.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Genome polishing is very important for long read assembly. This is due to the relatively high rate of error in PacBio and ONT technologies compared to Illumina.</p>
<p>The general principle is to align reads to the genome assembly to correct incorrect bases. This is carried out by determining the consensus on each base. For example:</p>
<ul>
<li>The 10th base in a contig is A.</li>
<li>Reads are aligned to the contig with 11 reads covering the 10th base.</li>
<li>Therefore there are 11 bases from the reads aligned to the 10th base.</li>
<li>4 of these bases are A, 1 is C, and 6 are Gs.</li>
<li>In this case the polisher may determine that the 10th base is more likely to be a G and so <strong>polishes</strong> the base to a G.</li>
</ul>
<p>If possible I would recommend polishing with Illumina reads instead of long reads due if they are available. This is because Illumina reads are more accurate. However, we only have PacBio reads so that is what we will use.</p>
<div id="racon" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Racon<a href="07-Polishing.html#racon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/racoon.png" width="20%" style="display: block; margin: auto;" /></p>
<p><code>Racon</code> is a polisher that can use Illumina, PacBio or ONT reads to polish genome assemblies. This makes it a lot more flexible that other polishers that can only use one or two types of sequencing technologies for polishing.</p>
<div id="racon-conda-environment-directory" class="section level3 hasAnchor" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> Racon: Conda environment &amp; directory<a href="07-Polishing.html#racon-conda-environment-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>bacterial_assembly</code> conda environment for our <code>Racon</code> polishing. Either:</p>
<ul>
<li>Use your currently open <code>bacterial_assembly</code> terminal.</li>
<li>Or, if you need a new <code>bacterial_assembly</code> terminal see the <a href="05-Assembly.html#redbeanassemblyconda">bacterial assembly conda activation instructions</a></li>
</ul>
<p>Ensure you are in the correct directory and create a new output directory.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="07-Polishing.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow</span>
<span id="cb22-2"><a href="07-Polishing.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> racon/redbean/ecoli</span></code></pre></div>
</div>
<div id="racon-index-assembly" class="section level3 hasAnchor" number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> Racon: Index assembly<a href="07-Polishing.html#racon-index-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/index.png" width="10%" style="display: block; margin: auto;" /></p>
<p>The first step is to <code>bwa index</code> the assembly we want polished. This will create various index files that start with <code>redbean_assembly/ecoli.ctg.fa</code> and end with different suffixes (<code>.ann</code>, <code>.bwt</code>, <code>.mmi</code>, etc).</p>
<p>Without the index files the read alignment process would be much more difficult and therefore slower. Imagine trying to search for a topic in an encyclopaedia without an index or trying to search for a topic in Wikipedia without the search bar. You could probably do it but it would take a lot longer.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="07-Polishing.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Index the assmebly</span></span>
<span id="cb23-2"><a href="07-Polishing.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> index redbean_assembly/ecoli.ctg.fa</span>
<span id="cb23-3"><a href="07-Polishing.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#List the index files</span></span>
<span id="cb23-4"><a href="07-Polishing.html#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> redbean_assembly/ecoli.ctg.fa.<span class="pp">*</span></span></code></pre></div>
</div>
<div id="racon-read-alignment-to-assembly" class="section level3 hasAnchor" number="7.1.3">
<h3><span class="header-section-number">7.1.3</span> Racon: Read alignment to assembly<a href="07-Polishing.html#racon-read-alignment-to-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/alignment.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With our index files created we can align the reads to the assembly. Ensure to use the reads which will be used for polishing. This may take &gt;10 minutes.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="07-Polishing.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="dt">\</span></span>
<span id="cb24-2"><a href="07-Polishing.html#cb24-2" aria-hidden="true" tabindex="-1"></a>-t 8 <span class="dt">\</span></span>
<span id="cb24-3"><a href="07-Polishing.html#cb24-3" aria-hidden="true" tabindex="-1"></a>-x pacbio <span class="dt">\</span></span>
<span id="cb24-4"><a href="07-Polishing.html#cb24-4" aria-hidden="true" tabindex="-1"></a>redbean_assembly/ecoli.ctg.fa <span class="dt">\</span></span>
<span id="cb24-5"><a href="07-Polishing.html#cb24-5" aria-hidden="true" tabindex="-1"></a>ecoli_reads/ecoli.fastq <span class="dt">\</span></span>
<span id="cb24-6"><a href="07-Polishing.html#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> racon/redbean/ecoli/ecoli_i1.sam</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-t</code> : Number of threads.</li>
<li><code>-x</code> : Read type. This will change various parameters to suit the read technology better.
<ul>
<li><code>pacbio</code> : Pacbio reads.</li>
<li><code>ont2d</code> : Oxford Nanopore 2D-reads</li>
</ul></li>
<li>The second last flagless input (<code>redbean_assembly/ecoli.ctg.fa</code>) indicates the indexed assembly the reads will be aligned to.</li>
<li>The last flagless input (<code>ecoli_reads/ecoli.fastq</code>) indicates the reads to be aligned.</li>
<li>The command would output the alignment to your screen (i.e. the standard output) but instead we will redirect (<code>&gt;</code>) it to a new file (<code>racon/redbean/ecoli/ecoli_i1.sam</code>). This is a SAM file which contains the alignment information (SAM file specification: <a href="https://samtools.github.io/hts-specs/SAMv1.pdf" class="uri">https://samtools.github.io/hts-specs/SAMv1.pdf</a>).</li>
</ul>
<p>BAM files are the binary version of SAM files. For most processes we would convert the SAM to BAM for storage purposes (BAM files are smaller than SAM files). However, <code>Racon</code> requires a SAM file so we will not convert it to BAM.</p>
</div>
<div id="racon-polishing" class="section level3 hasAnchor" number="7.1.4">
<h3><span class="header-section-number">7.1.4</span> Racon: Polishing<a href="07-Polishing.html#racon-polishing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/shiny_emerald.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Now that we have all the required files we will carry out <code>Racon</code> polishing.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="07-Polishing.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">racon</span> <span class="at">-t</span> 8 <span class="dt">\</span></span>
<span id="cb25-2"><a href="07-Polishing.html#cb25-2" aria-hidden="true" tabindex="-1"></a>ecoli_reads/ecoli.fastq <span class="dt">\</span></span>
<span id="cb25-3"><a href="07-Polishing.html#cb25-3" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i1.sam <span class="dt">\</span></span>
<span id="cb25-4"><a href="07-Polishing.html#cb25-4" aria-hidden="true" tabindex="-1"></a>redbean_assembly/ecoli.ctg.fa <span class="dt">\</span></span>
<span id="cb25-5"><a href="07-Polishing.html#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> racon/redbean/ecoli/ecoli_i1.fasta</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>-t</code> : Number of threads.</li>
<li>The first flagless input is the reads to be aligned.</li>
<li>The second flagless input is the alignment file (SAM file).</li>
<li>The third flagless input is the assembly to be polished.</li>
<li>The output is redirected (<code>&gt;</code>) to a fasta file (<code>racon/redbean/ecoli/ecoli_i1.fasta</code>). This contains the polished assembly.</li>
</ul>
</div>
<div id="racon-polishing-report" class="section level3 hasAnchor" number="7.1.5">
<h3><span class="header-section-number">7.1.5</span> Racon: Polishing report<a href="07-Polishing.html#racon-polishing-report" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/report.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Unfortunately <code>Racon</code> does not create a polishing report. We will therefore create our own to determine how much polishing was carried out. This can be carried out with the <code>dnadiff</code> command from the program <code>MUMmer</code>.</p>
<p><code>dnadiff</code> will create quite a few files so we will create an output directory for its results.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="07-Polishing.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> racon/redbean/ecoli/dnadiff</span></code></pre></div>
<p>We will use <code>dnadiff</code> to align the original assembly (reference) to our polished assembly (query). This will then provide detailed information on the differences between the two assemblies (i.e. we can see how polishing affected the assembly).</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="07-Polishing.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">dnadiff</span> <span class="dt">\</span></span>
<span id="cb27-2"><a href="07-Polishing.html#cb27-2" aria-hidden="true" tabindex="-1"></a>--prefix racon/redbean/ecoli/dnadiff/ref_redbean_query_raconi1 <span class="dt">\</span></span>
<span id="cb27-3"><a href="07-Polishing.html#cb27-3" aria-hidden="true" tabindex="-1"></a>redbean_assembly/ecoli.ctg.fa <span class="dt">\</span></span>
<span id="cb27-4"><a href="07-Polishing.html#cb27-4" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i1.fasta</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li>--prefix : Prefix of the output files. The output files will be in the directory <code>racon/redbean/ecoli/dnadiff/</code> with the prefix <code>ref_redbean_query_raconi1</code>.</li>
<li>The first flagless input is the reference assembly for comparison.</li>
<li>The second flagless input is the query assembly for comparison.</li>
</ul>
<p>List all the output files.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="07-Polishing.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> racon/redbean/ecoli/dnadiff</span></code></pre></div>
<p>There is a lot of information in them but for our purposes we will only inspect the report file.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="07-Polishing.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> racon/redbean/ecoli/dnadiff/ref_redbean_query_raconi1.report</span></code></pre></div>
<p>There is a lot of useful information in this file. The <strong>[REF]</strong> column represents the original assembly whilst the <strong>[QRY]</strong> column represent the polished assembly. Let us go through some (not all) of the stats.</p>
<p><strong>Note</strong>: Your values may differ from those mentioned below.</p>
<ul>
<li><strong>[Sequences]</strong>
<ul>
<li>TotalSeqs: We can see that we have dropped from 4 contigs to 1 contig. Racon has removed unpolished contigs.</li>
<li>AlignedSeqs: Additionally, The 4 contigs of the reference align to the 1 contig of the query and vice versa.</li>
</ul></li>
<li><strong>[Bases]</strong>
<ul>
<li>TotalBases: Our total bases has decreased (~50kbp) from reference to query.</li>
</ul></li>
<li><strong>[Feature Estimates]</strong>
<ul>
<li>Insertions: There are 19 insertion events that consists of ~50Kbp in the reference assembly. This indicates these areas were caused by misassembly and removed from the query by polishing.</li>
</ul></li>
<li><strong>[SNPs]</strong>
<ul>
<li>TotalSNPs: We can see there are 627 SNPs between the two assemblies. This shows 627bp were changed due to the polisher.</li>
<li>TotalIndels: 3782 single nucleotide Indels were found. This indicates a lot of bases were added and removed by polishing.
<ul>
<li>The first four rows in the [QRY] column show added bases. ~2500 bases were added by polishing.</li>
<li>The next four rows in the [QRY] column show bases removed by polishing. ~1200 bases were removed by polishing.</li>
</ul></li>
</ul></li>
</ul>
<p>This shows that polishing definitely worked and did quite a lot.</p>
<p>For more info on <code>dnadiff</code> and the report statistics README:
<a href="https://github.com/mummer4/mummer/blob/master/docs/dnadiff.README" class="uri">https://github.com/mummer4/mummer/blob/master/docs/dnadiff.README</a></p>
</div>
<div id="racon-second-iteration" class="section level3 hasAnchor" number="7.1.6">
<h3><span class="header-section-number">7.1.6</span> Racon: Second iteration<a href="07-Polishing.html#racon-second-iteration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/repeat.png" width="10%" style="display: block; margin: auto;" /></p>
<p>With genome polishing normally multiple iterations are required. Therefore let us polish the polished assembly to create the second iteration of polishing.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="07-Polishing.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Index iteration 1 polished assembly</span></span>
<span id="cb30-2"><a href="07-Polishing.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> index racon/redbean/ecoli/ecoli_i1.fasta</span>
<span id="cb30-3"><a href="07-Polishing.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Align the reads to the iteration 1 polished assembly</span></span>
<span id="cb30-4"><a href="07-Polishing.html#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">#This will create the second iteration SAM file</span></span>
<span id="cb30-5"><a href="07-Polishing.html#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="dt">\</span></span>
<span id="cb30-6"><a href="07-Polishing.html#cb30-6" aria-hidden="true" tabindex="-1"></a>-t 8 <span class="dt">\</span></span>
<span id="cb30-7"><a href="07-Polishing.html#cb30-7" aria-hidden="true" tabindex="-1"></a>-x pacbio <span class="dt">\</span></span>
<span id="cb30-8"><a href="07-Polishing.html#cb30-8" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i1.fasta <span class="dt">\</span></span>
<span id="cb30-9"><a href="07-Polishing.html#cb30-9" aria-hidden="true" tabindex="-1"></a>ecoli_reads/ecoli.fastq <span class="dt">\</span></span>
<span id="cb30-10"><a href="07-Polishing.html#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> racon/redbean/ecoli/ecoli_i2.sam</span>
<span id="cb30-11"><a href="07-Polishing.html#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Polish the iteration 1 polished assembly</span></span>
<span id="cb30-12"><a href="07-Polishing.html#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Carried out with the second iteration SAM file</span></span>
<span id="cb30-13"><a href="07-Polishing.html#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="ex">racon</span> <span class="dt">\</span></span>
<span id="cb30-14"><a href="07-Polishing.html#cb30-14" aria-hidden="true" tabindex="-1"></a>-t 8 <span class="dt">\</span></span>
<span id="cb30-15"><a href="07-Polishing.html#cb30-15" aria-hidden="true" tabindex="-1"></a>ecoli_reads/ecoli.fastq <span class="dt">\</span></span>
<span id="cb30-16"><a href="07-Polishing.html#cb30-16" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i2.sam <span class="dt">\</span></span>
<span id="cb30-17"><a href="07-Polishing.html#cb30-17" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i1.fasta <span class="dt">\</span></span>
<span id="cb30-18"><a href="07-Polishing.html#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="op">&gt;</span> racon/redbean/ecoli/ecoli_i2.fasta</span></code></pre></div>
<p>With the second iteration of polishing carried out we will assess how much polishing was carried out with <code>dnadiff</code>. In this case we will use the first iteration polished assembly as the reference and the second iteration polished assembly as the query.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="07-Polishing.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="ex">dnadiff</span> <span class="dt">\</span></span>
<span id="cb31-2"><a href="07-Polishing.html#cb31-2" aria-hidden="true" tabindex="-1"></a>--prefix racon/redbean/ecoli/dnadiff/ref_raconi1_query_raconi2 <span class="dt">\</span></span>
<span id="cb31-3"><a href="07-Polishing.html#cb31-3" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i1.fasta <span class="dt">\</span></span>
<span id="cb31-4"><a href="07-Polishing.html#cb31-4" aria-hidden="true" tabindex="-1"></a>racon/redbean/ecoli/ecoli_i2.fasta</span></code></pre></div>
<p>Inspect the new report file.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="07-Polishing.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> racon/redbean/ecoli/dnadiff/ref_raconi1_query_raconi2.report</span></code></pre></div>
</div>
<div id="racon-questions-summary" class="section level3 hasAnchor" number="7.1.7">
<h3><span class="header-section-number">7.1.7</span> Racon; Questions &amp; Summary<a href="07-Polishing.html#racon-questions-summary" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/sum_green.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Has the number of changes reduced?</p>
<p>Would you carry out further polishing?</p>
<p>The number of changes has indeed reduced by quite a bit, from 3782 to 40 TotalSNPs. When we are polishing we are looking for these stats to stop changing and converge to a similar value.</p>
<p>Unfortunately my experience is that no matter how many iterations you carry out these stats will not go to zero. Instead the polisher gets to a certain point where it cannot tell what base certain positions should be. As an example one position could be polished to change it from an A to a G then in the next polishing iteration it will change from G to A and so on.</p>
<p>So it would be advised to do some more polishing but for the sake of time we will continue on.</p>
</div>
</div>
<div id="other-polishers" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Other polishers<a href="07-Polishing.html#other-polishers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/pilon.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Other polishers to explore for yourself include (links in <a href="09-Annotation.html#manuals">appendix</a>):</p>
<ul>
<li><code>Pilon</code></li>
<li>PacBio's <code>GenomicConsensus</code></li>
<li>ONT's <code>Medaka</code></li>
<li><code>HyPo</code></li>
<li><code>NextPolish</code></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="06-Genome_assessment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="08-Circlator.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
