<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Annotation | Bacterial genome assembly and annotation</title>
  <meta name="description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Annotation | Bacterial genome assembly and annotation" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Annotation | Bacterial genome assembly and annotation" />
  
  <meta name="twitter:description" content="NEOF book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<<<<<<< HEAD
<meta name="date" content="2022-06-01" />
=======
<meta name="date" content="2022-06-08" />
>>>>>>> master

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="08-Circlator.html"/>
<link rel="next" href="10-Final_report.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="www/webex.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation</a></li>

<li class="divider"></li>
<li class="part"><span><b>Start</b></span></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_assembly_and_annotation.html"><a href="01-Bacterial_genome_assembly_and_annotation.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html"><i class="fa fa-check"></i><b>2</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>2.1</b> Logon instructions</a></li>
<li class="chapter" data-level="2.2" data-path="02-Cluster_Introduction.html"><a href="02-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>2.2</b> The Terminal Window</a></li>
</ul></li>
<li class="part"><span><b>Standard workflow</b></span></li>
<li class="chapter" data-level="3" data-path="03-Start.html"><a href="03-Start.html"><i class="fa fa-check"></i><b>3</b> Setup</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Start.html"><a href="03-Start.html#workshop-directory-data"><i class="fa fa-check"></i><b>3.1</b> Workshop directory &amp; data</a></li>
<li class="chapter" data-level="3.2" data-path="03-Start.html"><a href="03-Start.html#conda-environments"><i class="fa fa-check"></i><b>3.2</b> Conda environments</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Overview.html"><a href="04-Overview.html"><i class="fa fa-check"></i><b>4</b> Overview</a></li>
<li class="chapter" data-level="5" data-path="05-Assembly.html"><a href="05-Assembly.html"><i class="fa fa-check"></i><b>5</b> Genome Assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean"><i class="fa fa-check"></i><b>5.1</b> Redbean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="05-Assembly.html"><a href="05-Assembly.html#redbeanassemblyconda"><i class="fa fa-check"></i><b>5.1.1</b> Redbean: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="5.1.2" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-assemble-long-reads"><i class="fa fa-check"></i><b>5.1.2</b> Redbean: Assemble long reads</a></li>
<li class="chapter" data-level="5.1.3" data-path="05-Assembly.html"><a href="05-Assembly.html#redbean-derive-consensus"><i class="fa fa-check"></i><b>5.1.3</b> Redbean: Derive consensus</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="05-Assembly.html"><a href="05-Assembly.html#bandage-visualisation"><i class="fa fa-check"></i><b>5.2</b> Bandage Visualisation</a></li>
<li class="chapter" data-level="5.3" data-path="05-Assembly.html"><a href="05-Assembly.html#other-long-read-assemblers"><i class="fa fa-check"></i><b>5.3</b> Other long read assemblers</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html"><i class="fa fa-check"></i><b>6</b> Genome assembly assessment</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessmentconda"><i class="fa fa-check"></i><b>6.1</b> Assessment: Conda environnment &amp; directory</a></li>
<li class="chapter" data-level="6.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast"><i class="fa fa-check"></i><b>6.2</b> QUAST</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast-questions"><i class="fa fa-check"></i><b>6.2.1</b> QUAST Questions</a></li>
<li class="chapter" data-level="6.2.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#quast-summary"><i class="fa fa-check"></i><b>6.2.2</b> QUAST summary</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco"><i class="fa fa-check"></i><b>6.3</b> BUSCO</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-lineages"><i class="fa fa-check"></i><b>6.3.1</b> BUSCO lineages</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#run-busco"><i class="fa fa-check"></i><b>6.3.2</b> Run BUSCO</a></li>
<li class="chapter" data-level="6.3.3" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-output"><i class="fa fa-check"></i><b>6.3.3</b> BUSCO output</a></li>
<li class="chapter" data-level="6.3.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-questions"><i class="fa fa-check"></i><b>6.3.4</b> BUSCO questions</a></li>
<li class="chapter" data-level="6.3.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#busco-summary"><i class="fa fa-check"></i><b>6.3.5</b> BUSCO summary</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#checkm"><i class="fa fa-check"></i><b>6.4</b> CheckM</a></li>
<li class="chapter" data-level="6.5" data-path="06-Genome_assessment.html"><a href="06-Genome_assessment.html#assessment-recap"><i class="fa fa-check"></i><b>6.5</b> Assessment recap</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Polishing.html"><a href="07-Polishing.html"><i class="fa fa-check"></i><b>7</b> Polishing</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon"><i class="fa fa-check"></i><b>7.1</b> Racon</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-conda-environment-directory"><i class="fa fa-check"></i><b>7.1.1</b> Racon: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="7.1.2" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-index-assembly"><i class="fa fa-check"></i><b>7.1.2</b> Racon: Index assembly</a></li>
<li class="chapter" data-level="7.1.3" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-read-alignment-to-assembly"><i class="fa fa-check"></i><b>7.1.3</b> Racon: Read alignment to assembly</a></li>
<li class="chapter" data-level="7.1.4" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing"><i class="fa fa-check"></i><b>7.1.4</b> Racon: Polishing</a></li>
<li class="chapter" data-level="7.1.5" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-polishing-report"><i class="fa fa-check"></i><b>7.1.5</b> Racon: Polishing report</a></li>
<li class="chapter" data-level="7.1.6" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-second-iteration"><i class="fa fa-check"></i><b>7.1.6</b> Racon: Second iteration</a></li>
<li class="chapter" data-level="7.1.7" data-path="07-Polishing.html"><a href="07-Polishing.html#racon-questions-summary"><i class="fa fa-check"></i><b>7.1.7</b> Racon; Questions &amp; Summary</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="07-Polishing.html"><a href="07-Polishing.html#other-polishers"><i class="fa fa-check"></i><b>7.2</b> Other polishers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Circlator.html"><a href="08-Circlator.html"><i class="fa fa-check"></i><b>8</b> Circularisation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator"><i class="fa fa-check"></i><b>8.1</b> Circlator</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-conda-environment-directory"><i class="fa fa-check"></i><b>8.1.1</b> Circlator: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="8.1.2" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-read-correction"><i class="fa fa-check"></i><b>8.1.2</b> Circlator: Read correction</a></li>
<li class="chapter" data-level="8.1.3" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-circularisation"><i class="fa fa-check"></i><b>8.1.3</b> Circlator: Circularisation</a></li>
<li class="chapter" data-level="8.1.4" data-path="08-Circlator.html"><a href="08-Circlator.html#circlator-assessment-polishing"><i class="fa fa-check"></i><b>8.1.4</b> Circlator: Assessment &amp; Polishing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Annotation.html"><a href="09-Annotation.html"><i class="fa fa-check"></i><b>9</b> Annotation</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka"><i class="fa fa-check"></i><b>9.1</b> Prokka</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-conda-environment-directory"><i class="fa fa-check"></i><b>9.1.1</b> Prokka: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="9.1.2" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-run"><i class="fa fa-check"></i><b>9.1.2</b> Prokka: Run</a></li>
<li class="chapter" data-level="9.1.3" data-path="09-Annotation.html"><a href="09-Annotation.html#prokka-output"><i class="fa fa-check"></i><b>9.1.3</b> Prokka: Output</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="09-Annotation.html"><a href="09-Annotation.html#exploring-annotations"><i class="fa fa-check"></i><b>9.2</b> Exploring annotations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Final_report.html"><a href="10-Final_report.html"><i class="fa fa-check"></i><b>10</b> Final report</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report"><i class="fa fa-check"></i><b>10.1</b> Report</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="10-Final_report.html"><a href="10-Final_report.html#report-conda-environment-directory"><i class="fa fa-check"></i><b>10.1.1</b> Report: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="10.1.2" data-path="10-Final_report.html"><a href="10-Final_report.html#report-quast"><i class="fa fa-check"></i><b>10.1.2</b> Report: QUAST</a></li>
<li class="chapter" data-level="10.1.3" data-path="10-Final_report.html"><a href="10-Final_report.html#report-busco"><i class="fa fa-check"></i><b>10.1.3</b> Report: BUSCO</a></li>
<li class="chapter" data-level="10.1.4" data-path="10-Final_report.html"><a href="10-Final_report.html#report-multiqc"><i class="fa fa-check"></i><b>10.1.4</b> Report: MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="10-Final_report.html"><a href="10-Final_report.html#final-recap"><i class="fa fa-check"></i><b>10.2</b> Final recap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Cleanup.html"><a href="11-Cleanup.html"><i class="fa fa-check"></i><b>11</b> Cleanup</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Cleanup.html"><a href="11-Cleanup.html#deleting-unwanted-data"><i class="fa fa-check"></i><b>11.1</b> Deleting unwanted data</a></li>
<li class="chapter" data-level="11.2" data-path="11-Cleanup.html"><a href="11-Cleanup.html#compressing-the-final-directory"><i class="fa fa-check"></i><b>11.2</b> Compressing the final directory</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="12-Appendix.html"><a href="12-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="09-Annotation.html"><a href="09-Annotation.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="12-Appendix.html"><a href="12-Appendix.html#obtaining-read-data"><i class="fa fa-check"></i><b>C</b> Obtaining Read Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="annotation" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Annotation<a href="09-Annotation.html#annotation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figures/gene_anno.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Genome annotation is the act of identifying genomics features and adding useful information to these annotations. This includes adding information on genes, repeat regions, pathways etc..</p>
<div id="prokka" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Prokka<a href="09-Annotation.html#prokka" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/quokka.png" width="15%" style="display: block; margin: auto;" /></p>
<p><code>Prokka</code> is a rapid genome annotation tool for bacterial, archaeal, and viral genomes. It is the standard method for bacterial genome annotation and wraps various other tools to carry out its processes.</p>
<p><code>Prokka</code> is fast, reliable and basic. It will detect genomic features such as CDS (coding sequence), genes, rRNA, and repeat regions. Additionally, it will provide pathway information for CDS annotations in the form of:</p>
<ul>
<li><strong>EC (Enzyme Commission) numbers</strong>: This is a scheme of numbers grouping enzymes based on the reaction they catalyse.
<ul>
<li>An example EC number is <strong>EC 3.1.1</strong>. The numbers are hierarchical.
<ul>
<li>EC 3 = Hydrolase enzymes.</li>
<li>EC 3.1 = Hydrolase enzymes that act on ester bonds.</li>
<li>EC 3.1.1 = Carboxylic Ester Hydrolases.</li>
</ul></li>
<li>More info : <a href="https://en.wikipedia.org/wiki/Enzyme_Commission_number" class="uri">https://en.wikipedia.org/wiki/Enzyme_Commission_number</a></li>
</ul></li>
<li><strong>COG</strong>: Cluster of Orthologous Genes. Orthologs are gene sequences derived from the same ancestral gene present in two species' last common ancestor. COGs group CDS into three main tiers.
<ol style="list-style-type: decimal">
<li>COG Categories: This is the highest level grouping and are represented by the alphabet. Examples include:
<ul>
<li>K - Transcription</li>
<li>Z - Cytoskeleton</li>
<li>L - Replication, Recombination, and Repair</li>
</ul></li>
<li>Pathways: The step below categories. There are 63 different pathways represented in COG. Each pathway represents a different number of COGs.
<ul>
<li>Examples of COG pathways include 16S rRNA modifciation (15 COGs), Arginine biosynthesis (12 COGs), and Glycolysis (18 COGs).</li>
<li>List of COG pathways: <a href="https://www.ncbi.nlm.nih.gov/research/cog/pathways/" class="uri">https://www.ncbi.nlm.nih.gov/research/cog/pathways/</a></li>
</ul></li>
<li>COGs: These are the clusters of Orthologous Genes.
<ul>
<li>COGs are represented by a number, a symbol, and a name.</li>
</ul></li>
</ol></li>
</ul>
<table>
<caption>Examples of COGs and their hierarchical classifications</caption>
<colgroup>
<col width="30%" />
<col width="12%" />
<col width="6%" />
<col width="6%" />
<col width="45%" />
</colgroup>
<thead>
<tr class="header">
<th>COG category</th>
<th>Pathway</th>
<th>COG number</th>
<th>COG symbol</th>
<th>Name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>J - Translation, Ribosomal Structure, and Biogenesis</td>
<td>16S rRNA modification</td>
<td>COG0742</td>
<td>RsmD</td>
<td>16S rRNA G966 N2-methylase RsmD</td>
</tr>
<tr class="even">
<td>K - Transcription &amp; E - Amino Acid Transport &amp; Metabolism</td>
<td>Lysine biosynthesis</td>
<td>COG1167</td>
<td>ARO8</td>
<td>DNA-binding transcriptional regulator, MocR family, contains an aminotransferase domain</td>
</tr>
<tr class="odd">
<td>H - Coenzyme transport and Metabolism</td>
<td>Folate biosynthesis</td>
<td>COG0302</td>
<td>FolE</td>
<td>GTP cyclohydrolase I</td>
</tr>
</tbody>
</table>
<div id="prokka-conda-environment-directory" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Prokka: Conda environment &amp; directory<a href="09-Annotation.html#prokka-conda-environment-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Open a new terminal (right click on the main screen, choose <code>Applications</code> -&gt; <code>Shell</code> -&gt; <code>bash</code>).</p>
<p>We will use the <code>prokka</code> <code>conda</code> environment for our genome annotation.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="09-Annotation.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="ex">chos</span> 8</span>
<span id="cb39-2"><a href="09-Annotation.html#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> useprokka</span></code></pre></div>
<p>Ensure you are in the standard workflow directory.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb40-1"><a href="09-Annotation.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/standard_workflow</span></code></pre></div>
<p>Whenever I have my final genome assemblies I like to put them into a final directory that will contain all the output for them. Create a final directory and move into it.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="09-Annotation.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> final_assembly</span>
<span id="cb41-2"><a href="09-Annotation.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> final_assembly</span></code></pre></div>
<p>For ease I also like to make a directory that contains all the final assemblies. Additionally, it allows us to name the fasta files with the sample name.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="09-Annotation.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make directory</span></span>
<span id="cb42-2"><a href="09-Annotation.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> assembly</span>
<span id="cb42-3"><a href="09-Annotation.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Copy final assembly naming it with the sample name (ecoli)</span></span>
<span id="cb42-4"><a href="09-Annotation.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../circlator/redbean_racon_i2/ecoli/06.fixstart.fasta assembly/ecoli.fasta</span></code></pre></div>
</div>
<div id="prokka-run" class="section level3 hasAnchor" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Prokka: Run<a href="09-Annotation.html#prokka-run" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/enter_key.png" width="10%" style="display: block; margin: auto;" /></p>
<p>Now it is time to run <code>Prokka</code> annotation</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="09-Annotation.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Make a prokka directory</span></span>
<span id="cb43-2"><a href="09-Annotation.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> prokka</span>
<span id="cb43-3"><a href="09-Annotation.html#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Run Prokka</span></span>
<span id="cb43-4"><a href="09-Annotation.html#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="co">#outdir must not exist</span></span>
<span id="cb43-5"><a href="09-Annotation.html#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="ex">prokka</span> <span class="dt">\</span></span>
<span id="cb43-6"><a href="09-Annotation.html#cb43-6" aria-hidden="true" tabindex="-1"></a>--addgenes <span class="dt">\</span></span>
<span id="cb43-7"><a href="09-Annotation.html#cb43-7" aria-hidden="true" tabindex="-1"></a>--prefix ecoli <span class="dt">\</span></span>
<span id="cb43-8"><a href="09-Annotation.html#cb43-8" aria-hidden="true" tabindex="-1"></a>--genus Escherichia <span class="dt">\</span></span>
<span id="cb43-9"><a href="09-Annotation.html#cb43-9" aria-hidden="true" tabindex="-1"></a>--species coli <span class="dt">\</span></span>
<span id="cb43-10"><a href="09-Annotation.html#cb43-10" aria-hidden="true" tabindex="-1"></a>--kingdom Bacteria <span class="dt">\</span></span>
<span id="cb43-11"><a href="09-Annotation.html#cb43-11" aria-hidden="true" tabindex="-1"></a>--outdir prokka/ecoli <span class="dt">\</span></span>
<span id="cb43-12"><a href="09-Annotation.html#cb43-12" aria-hidden="true" tabindex="-1"></a>assembly/ecoli.fasta</span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>--addgenes</code> : This will add gene features for each CDS feature.</li>
<li><code>--prefix</code> : The prefix of the output files.</li>
<li><code>--genus</code> : The genus name that will be included in output.</li>
<li><code>--species</code> : The species name that will be included in output.</li>
<li><code>--kingdom</code> : Annotation mode to be used. Options are:
<ul>
<li><code>Bacteria</code> (default)</li>
<li><code>Archaea</code></li>
<li><code>Mitochondria</code></li>
<li><code>Viruses</code></li>
</ul></li>
<li><code>--outdir</code> : The output directory of <code>Prokka</code>. This must not exist before running the command.</li>
<li>The last flagless parameter is the fasta file to be annotated.</li>
</ul>
</div>
<div id="prokka-output" class="section level3 hasAnchor" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> Prokka: Output<a href="09-Annotation.html#prokka-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="figures/output.png" width="10%" style="display: block; margin: auto;" /></p>
<p>There are a lot of output files in the output directory. Below is a table with information on all these files.</p>
<table>
<caption>From: <a href="https://github.com/tseemann/prokka" class="uri">https://github.com/tseemann/prokka</a></caption>
<colgroup>
<col width="7%" />
<col width="92%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Extension</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>.gff</td>
<td>This is the master annotation in GFF3 format, containing both sequences and annotations. It can be viewed directly in Artemis or IGV.</td>
</tr>
<tr class="even">
<td>.gbk</td>
<td>This is a standard Genbank file derived from the master .gff. If the input to prokka was a multi-FASTA, then this will be a multi-Genbank, with one record for each sequence.</td>
</tr>
<tr class="odd">
<td>.fna</td>
<td>Nucleotide FASTA file of the input contig sequences.</td>
</tr>
<tr class="even">
<td>.faa</td>
<td>Protein FASTA file of the translated CDS sequences.</td>
</tr>
<tr class="odd">
<td>.ffn</td>
<td>Nucleotide FASTA file of all the prediction transcripts (CDS, rRNA, tRNA, tmRNA, misc_RNA)</td>
</tr>
<tr class="even">
<td>.sqn</td>
<td>An ASN1 format "Sequin" file for submission to Genbank. It needs to be edited to set the correct taxonomy, authors, related publication etc.</td>
</tr>
<tr class="odd">
<td>.fsa</td>
<td>Nucleotide FASTA file of the input contig sequences, used by "tbl2asn" to create the .sqn file. It is mostly the same as the .fna file, but with extra Sequin tags in the sequence description lines.</td>
</tr>
<tr class="even">
<td>.tbl</td>
<td>Feature Table file, used by "tbl2asn" to create the .sqn file.</td>
</tr>
<tr class="odd">
<td>.err</td>
<td>Unacceptable annotations - the NCBI discrepancy report.</td>
</tr>
<tr class="even">
<td>.log</td>
<td>Contains all the output that Prokka produced during its run. This is a record of what settings you used, even if the --quiet option was enabled.</td>
</tr>
<tr class="odd">
<td>.txt</td>
<td>Statistics relating to the annotated features found.</td>
</tr>
<tr class="even">
<td>.tsv</td>
<td>Tab-separated file of all features: locus_tag,ftype,len_bp,gene,EC_number,COG,product</td>
</tr>
</tbody>
</table>
<p>The first file to look at is the summary <code>.txt</code> file.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="09-Annotation.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> prokka/ecoli/ecoli.txt</span></code></pre></div>
<p><strong>Questions</strong></p>
<ul>
<li>How many CDS are annotated?</li>
<li>How many repeat regions were detected?</li>
</ul>
<p>The next best human-readable file is the <code>.tsv</code> file. This is in essence a table with one row per annotated feature.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="09-Annotation.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> <span class="at">-S</span> prokka/ecoli/ecoli.tsv</span></code></pre></div>
<p><strong>Questions</strong></p>
<ul>
<li>What type of feature (ftype) is the first annotated feature?</li>
<li>Which have more information tied to them, CDS or gene features?</li>
<li>How many CDS features have been annotated as "kefF"? (Tip: The <code>grep</code> command is useful)</li>
<li>What is the COG name for COG1662?</li>
</ul>
<p>We have our final assembly and our annotation now. In the next step we wll create a final report.</p>
</div>
</div>
<div id="exploring-annotations" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Exploring annotations<a href="09-Annotation.html#exploring-annotations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="figures/explore.png" width="10%" style="display: block; margin: auto;" /></p>
<p>There are various ways to inspect the annotation data. We will not go into them here, instead just introducing you to some tools and databases. Links to the tools are in the <span id="manuals">Manuals</span> section of the appendix.</p>
<ul>
<li><code>IGV</code></li>
<li><code>Artemis</code></li>
<li><code>GhostKoala</code></li>
<li><code>MinPath</code></li>
</ul>

</div>
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  console.log("webex: update total_correct");

  if (t = document.getElementById("webex-total_correct")) {
    var correct = document.getElementsByClassName("webex-correct").length;
    var solvemes = document.getElementsByClassName("webex-solveme").length;
    var radiogroups = document.getElementsByClassName("webex-radiogroup").length;
    var selects = document.getElementsByClassName("webex-select").length;
    
    t.innerHTML = correct + " of " + (solvemes + radiogroups + selects) + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  console.log("webex: toggle hide");
  
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  console.log("webex: check solveme");

  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }

  if (my_answer == "") {
    cl.remove("webex-correct");
    cl.remove("webex-incorrect");
  } else if (real_answers.includes(my_answer)) {
    cl.add("webex-correct");
    cl.remove("webex-incorrect");
  } else {
    cl.add("webex-incorrect");
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }
  }

  update_total_correct();
}

/* function for checking select answers */
select_func = function(e) {
  console.log("webex: check select");
  
  var cl = this.classList
  
  /* add style */
  cl.remove("webex-incorrect");
  cl.remove("webex-correct");
  if (this.value == "answer") {
    cl.add("webex-correct");
  } else if (this.value != "blank") {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

/* function for checking radiogroups answers */
radiogroups_func = function(e) {
  console.log("webex: check radiogroups");

  var checked_button = document.querySelector('input[name=' + this.id + ']:checked');
  var cl = checked_button.parentElement.classList;
  var labels = checked_button.parentElement.parentElement.children;
  
  /* get rid of styles */
  for (i = 0; i < labels.length; i++) {
    labels[i].classList.remove("webex-incorrect");
    labels[i].classList.remove("webex-correct");
  }
  
  /* add style */
  if (checked_button.value == "answer") {
    cl.add("webex-correct");
  } else {
    cl.add("webex-incorrect");
  }
  
  update_total_correct();
}

window.onload = function() {
  console.log("onload");
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }

  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off");
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";

    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;

    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  /* set up radiogroups */
  var radiogroups = document.getElementsByClassName("webex-radiogroup");
  for (var i = 0; i < radiogroups.length; i++) {
    radiogroups[i].onchange = radiogroups_func;
  }
  
  /* set up selects */
  var selects = document.getElementsByClassName("webex-select");
  for (var i = 0; i < selects.length; i++) {
    selects[i].onchange = select_func;
  }

  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="08-Circlator.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="10-Final_report.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
