<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Hybrid assembly | Bacterial genome assembly and annotation supplemental</title>
  <meta name="description" content="NEOF supplemental book for the Bacterial genome assembly and annotation workshop" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Hybrid assembly | Bacterial genome assembly and annotation supplemental" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="/figures/NEOF.png" />
  <meta property="og:description" content="NEOF supplemental book for the Bacterial genome assembly and annotation workshop" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Hybrid assembly | Bacterial genome assembly and annotation supplemental" />
  
  <meta name="twitter:description" content="NEOF supplemental book for the Bacterial genome assembly and annotation workshop" />
  <meta name="twitter:image" content="/figures/NEOF.png" />

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2021-08-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="figures/NEOF_favicon.png" type="image/x-icon" />
<link rel="prev" href="03-High_coverage.html"/>
<link rel="next" href="05-Appendix.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="https://neof.org.uk/" target="blank"><img src="figures/neof_small_logo.png"></a></li>
<li><a>Bacterial genome assembly and annotation supplemental</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_genome_supplemental.html"><a href="01-Bacterial_genome_supplemental.html"><i class="fa fa-check"></i><b>1</b> Supplemental</a></li>
<li class="chapter" data-level="2" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html"><i class="fa fa-check"></i><b>2</b> Multiple sample workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html#multiple-sample-raw-data"><i class="fa fa-check"></i><b>2.1</b> Multiple sample raw data</a></li>
<li class="chapter" data-level="2.2" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html#sample-file"><i class="fa fa-check"></i><b>2.2</b> Sample file</a></li>
<li class="chapter" data-level="2.3" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html#start-looping"><i class="fa fa-check"></i><b>2.3</b> Start looping</a></li>
<li class="chapter" data-level="2.4" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html#continuing-the-workflow"><i class="fa fa-check"></i><b>2.4</b> Continuing the workflow</a></li>
<li class="chapter" data-level="2.5" data-path="02-Multi-sample_workflow.html"><a href="02-Multi-sample_workflow.html#gnu-parallel"><i class="fa fa-check"></i><b>2.5</b> GNU Parallel</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-High_coverage.html"><a href="03-High_coverage.html"><i class="fa fa-check"></i><b>3</b> High coverage</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-High_coverage.html"><a href="03-High_coverage.html#detecting-high-coverage"><i class="fa fa-check"></i><b>3.1</b> Detecting high coverage</a></li>
<li class="chapter" data-level="3.2" data-path="03-High_coverage.html"><a href="03-High_coverage.html#subsampling"><i class="fa fa-check"></i><b>3.2</b> Subsampling</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Hybrid_assembly.html"><a href="04-Hybrid_assembly.html"><i class="fa fa-check"></i><b>4</b> Hybrid assembly</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Hybrid_assembly.html"><a href="04-Hybrid_assembly.html#hybrid-assembly-conda-environment-directory"><i class="fa fa-check"></i><b>4.1</b> Hybrid assembly: Conda environment &amp; directory</a></li>
<li class="chapter" data-level="4.2" data-path="04-Hybrid_assembly.html"><a href="04-Hybrid_assembly.html#hybridspades"><i class="fa fa-check"></i><b>4.2</b> hybridSPAdes</a></li>
<li class="chapter" data-level="4.3" data-path="04-Hybrid_assembly.html"><a href="04-Hybrid_assembly.html#pilon"><i class="fa fa-check"></i><b>4.3</b> Pilon</a></li>
<li class="chapter" data-level="4.4" data-path="04-Hybrid_assembly.html"><a href="04-Hybrid_assembly.html#hybrid-approaches-recap"><i class="fa fa-check"></i><b>4.4</b> Hybrid approaches recap</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="05-Appendix.html"><a href="05-Appendix.html"><i class="fa fa-check"></i><b>A</b> Next steps</a></li>
<li class="chapter" data-level="B" data-path="05-Appendix.html"><a href="05-Appendix.html#manuals"><i class="fa fa-check"></i><b>B</b> Manuals</a></li>
<li class="chapter" data-level="C" data-path="05-Appendix.html"><a href="05-Appendix.html#data"><i class="fa fa-check"></i><b>C</b> Data</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial genome assembly and annotation supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hybrid" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Hybrid assembly</h1>
<p><img src="figures/hybrid_car.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Hybrid assembly methods can be used when you have long read (PacBio or ONT) and short read (Illumina) sequencing data for your samples. There are two main methods for this.</p>
<ol style="list-style-type: decimal">
<li>Using both short read and long read data during the assembly step (<code>hybridSPAdes</code>)</li>
<li>Assemble with the long reads and polish with the short reads (<code>pilon</code>)</li>
</ol>
<div id="hybrid-assembly-conda-environment-directory" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Hybrid assembly: Conda environment &amp; directory</h2>
<p><img src="figures/conda.png" width="10%" style="display: block; margin: auto;" /></p>
<p>We will use the <code>hybrid</code> <code>conda</code> environment for our hybrid assembly methods.</p>
<p>Activate the environment:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="04-Hybrid_assembly.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">chos</span> 8</span>
<span id="cb13-2"><a href="04-Hybrid_assembly.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> usehybrid</span></code></pre></div>
<p>Before running any commands we will get the Illumina and PacBio data directories setup.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="04-Hybrid_assembly.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Change directory to bacterial_assembly directory</span></span>
<span id="cb14-2"><a href="04-Hybrid_assembly.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/</span>
<span id="cb14-3"><a href="04-Hybrid_assembly.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Make a hybrid analysis directory</span></span>
<span id="cb14-4"><a href="04-Hybrid_assembly.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> hybrid_methods</span>
<span id="cb14-5"><a href="04-Hybrid_assembly.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Move into the hybrid_methods directory</span></span>
<span id="cb14-6"><a href="04-Hybrid_assembly.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> hybrid_methods</span>
<span id="cb14-7"><a href="04-Hybrid_assembly.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Make directories to contain softlinks for Illumina and PacBio data</span></span>
<span id="cb14-8"><a href="04-Hybrid_assembly.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> illumina_data</span>
<span id="cb14-9"><a href="04-Hybrid_assembly.html#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> pacbio_data</span>
<span id="cb14-10"><a href="04-Hybrid_assembly.html#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Make softlinks of E.coli sequencing data</span></span>
<span id="cb14-11"><a href="04-Hybrid_assembly.html#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub39/tea/nsc006/NEOF/bact_assembly/illumina_data/<span class="pp">*</span> illumina_data</span>
<span id="cb14-12"><a href="04-Hybrid_assembly.html#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub39/tea/nsc006/NEOF/bact_assembly/data/ecoli.fastq pacbio_data</span></code></pre></div>
</div>
<div id="hybridspades" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> hybridSPAdes</h2>
<p><img src="figures/spades.png" width="15%" style="display: block; margin: auto;" /></p>
<p>A popular hybrid assembler for bacterial genomes is <code>hybridSPAdes</code>. It’s use case is very similar to just using Illumina data. The only difference is the addition of the <code>--pacbio</code> for PacBio reads or <code>--nanopore</code> for Oxford nanopore reads.</p>
<p>First we will create an output directory for <code>HybridSPAdes</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="04-Hybrid_assembly.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Main output directory</span></span>
<span id="cb15-2"><a href="04-Hybrid_assembly.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> hybridspades</span>
<span id="cb15-3"><a href="04-Hybrid_assembly.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Ouptut directory for our sample</span></span>
<span id="cb15-4"><a href="04-Hybrid_assembly.html#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> hybridspades/ecoli</span></code></pre></div>
<p>Now to run <code>HybridSPAdes</code> with our Illumina and PacBio data.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="04-Hybrid_assembly.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">spades.py</span> <span class="dt">\</span></span>
<span id="cb16-2"><a href="04-Hybrid_assembly.html#cb16-2" aria-hidden="true" tabindex="-1"></a>-1 illumina_data/ecoli_R1.fastq <span class="dt">\</span></span>
<span id="cb16-3"><a href="04-Hybrid_assembly.html#cb16-3" aria-hidden="true" tabindex="-1"></a>-2 illumina_data/ecoli_R2.fastq <span class="dt">\</span></span>
<span id="cb16-4"><a href="04-Hybrid_assembly.html#cb16-4" aria-hidden="true" tabindex="-1"></a>--pacbio pacbio_data/ecoli.fastq <span class="dt">\</span></span>
<span id="cb16-5"><a href="04-Hybrid_assembly.html#cb16-5" aria-hidden="true" tabindex="-1"></a>-o hybridspades/ecoli <span class="dt">\</span></span>
<span id="cb16-6"><a href="04-Hybrid_assembly.html#cb16-6" aria-hidden="true" tabindex="-1"></a>-t 8 <span class="at">-m</span> 50</span></code></pre></div>
<p><strong>Note</strong>: This command will take about 10 minutes.</p>
<p><strong>Parameters</strong>:</p>
<ul>
<li><code>-1</code> : The forward paired-end reads.</li>
<li><code>-2</code> : The reverse paired-end reads.</li>
<li><code>--pacbio</code> : Provide PacBio reads for a hybrid assembly.
<ul>
<li><code>--nanopore</code> : Used for ONT data.</li>
</ul></li>
<li><code>-o</code> : The output directory.</li>
<li><code>-t</code> : The number of threads to be used with the process (default is 16).</li>
<li><code>-m</code> : The max RAM (in Gb) to be used for the assembly (default is 250Gb).</li>
</ul>
<p><code>SPAdes</code> can be very memory (RAM) hungry. This can be dangerous as using too much RAM is an easy way to crash the machine you are using. The flag <code>-m</code> is very good to prevent this . Additionally, decreasing the number of threads (<code>-t</code>) will decrease the RAM load.</p>
<p>Next view the output directory of this command.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="04-Hybrid_assembly.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> hybridspades/ecoli</span></code></pre></div>
<p>You will notice a lot of output. Most of this can be deleted (especially the directories starting with <code>K</code>). The actually assembly in fasta format is the file <code>contigs.fasta</code>.</p>
<p>For more info on all the output please see: <a href="https://github.com/ablab/spades#spadesoutsec" class="uri">https://github.com/ablab/spades#spadesoutsec</a></p>
</div>
<div id="pilon" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Pilon</h2>
<p><img src="figures/pilon.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Another method for a hybrid approach is to carry out the standard workflow using only the long reads and then polish with short (Illumina) reads.</p>
<p>In this case we are going to carry out polishing using the tool <code>Pilon</code>. In theory assembling with the long reads is good to produce a very continuous assembly and then polishing with Illumina reads should drastically improve the quality and accuracy.</p>
<p>First we will make a directory and make a softlink of the circularised genome assembly from the standard workflow.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="04-Hybrid_assembly.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Move to main directory</span></span>
<span id="cb18-2"><a href="04-Hybrid_assembly.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Change directory to bacterial_assembly directory</span></span>
<span id="cb18-3"><a href="04-Hybrid_assembly.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/bacterial_assembly/hybrid_methods</span>
<span id="cb18-4"><a href="04-Hybrid_assembly.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Make assembly for long read assembly</span></span>
<span id="cb18-5"><a href="04-Hybrid_assembly.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> long_read_assembly</span>
<span id="cb18-6"><a href="04-Hybrid_assembly.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Softlink of assembly</span></span>
<span id="cb18-7"><a href="04-Hybrid_assembly.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Using a premade file in case you don&#39;t have yours</span></span>
<span id="cb18-8"><a href="04-Hybrid_assembly.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /pub39/tea/nsc006/NEOF/bact_assembly/standard_workflow/ecoli.fasta long_read_assembly</span></code></pre></div>
<p>The next part is to align our Illumina reads to our assembly. This is similar to the process for <code>Racon</code>.</p>
<p>First we index the assembly.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="04-Hybrid_assembly.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> index long_read_assembly/ecoli.fasta</span></code></pre></div>
<p>Next we align our reads to the assembly we want polished. We carry this out with <code>bwa</code> which aligns Illumina reads by default.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="04-Hybrid_assembly.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create output directory</span></span>
<span id="cb20-2"><a href="04-Hybrid_assembly.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> pilon/ecoli</span>
<span id="cb20-3"><a href="04-Hybrid_assembly.html#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Align the reads</span></span>
<span id="cb20-4"><a href="04-Hybrid_assembly.html#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="dt">\</span></span>
<span id="cb20-5"><a href="04-Hybrid_assembly.html#cb20-5" aria-hidden="true" tabindex="-1"></a>long_read_assembly/ecoli.fasta <span class="dt">\</span></span>
<span id="cb20-6"><a href="04-Hybrid_assembly.html#cb20-6" aria-hidden="true" tabindex="-1"></a>illumina_data/ecoli_R1.fastq <span class="dt">\</span></span>
<span id="cb20-7"><a href="04-Hybrid_assembly.html#cb20-7" aria-hidden="true" tabindex="-1"></a>illumina_data/ecoli_R2.fastq <span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb20-8"><a href="04-Hybrid_assembly.html#cb20-8" aria-hidden="true" tabindex="-1"></a>pilon/ecoli/aln_i1.sam</span></code></pre></div>
<p><strong>Note</strong>: We have included <code>i1</code> in the output file to indicate this is the first polishing iteration.</p>
<p><code>Pilon</code> requires an indexed sorted <code>BAM</code> file. We will use <code>samtools</code> to produce this from our <code>SAM</code> file.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="04-Hybrid_assembly.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create a bam file from the sam file</span></span>
<span id="cb21-2"><a href="04-Hybrid_assembly.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-Sb</span> pilon/ecoli/aln_i1.sam <span class="op">&gt;</span> pilon/ecoli/aln_i1.bam</span>
<span id="cb21-3"><a href="04-Hybrid_assembly.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Delete the old sam file</span></span>
<span id="cb21-4"><a href="04-Hybrid_assembly.html#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> pilon/ecoli/aln_i1.sam</span>
<span id="cb21-5"><a href="04-Hybrid_assembly.html#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Sort the bam file</span></span>
<span id="cb21-6"><a href="04-Hybrid_assembly.html#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort pilon/ecoli/aln_i1.bam <span class="at">-o</span> pilon/ecoli/aln_i1.sort.bam</span>
<span id="cb21-7"><a href="04-Hybrid_assembly.html#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co">#delete the original bam file</span></span>
<span id="cb21-8"><a href="04-Hybrid_assembly.html#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> pilon/ecoli/aln_i1.bam</span>
<span id="cb21-9"><a href="04-Hybrid_assembly.html#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Index the sorted bam file</span></span>
<span id="cb21-10"><a href="04-Hybrid_assembly.html#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index pilon/ecoli/aln_i1.sort.bam</span></code></pre></div>
<p>Now we have a sorted <code>.bam</code> file and a <code>.bam.bai</code> file that contains the index information.</p>
<p>With this we can carry out <code>Pilon</code> polishing. <code>Pilon</code> is a <code>java</code> based program so we need to set the max memory (RAM) outside of the actual command. We’ll assign the max RAM of all future java based programs within the current terminal (including <code>Pilon</code>) to 30 Gb.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="04-Hybrid_assembly.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">_JAVA_OPTIONS=</span><span class="st">&quot;-Xmx30g&quot;</span></span></code></pre></div>
<p><strong>Yes</strong>, the <code>_</code> before <code>JAVA</code> needs to be there. For more info on Java heap sizes check out: <a href="https://alvinalexander.com/blog/post/java/java-xmx-xms-memory-heap-size-control/" class="uri">https://alvinalexander.com/blog/post/java/java-xmx-xms-memory-heap-size-control/</a></p>
<p>With the maximum memory set to 30Gb (the default size is too low) we can run our first iteration of <code>Pilon</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="04-Hybrid_assembly.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">pilon</span> <span class="dt">\</span></span>
<span id="cb23-2"><a href="04-Hybrid_assembly.html#cb23-2" aria-hidden="true" tabindex="-1"></a>--genome long_read_assembly/ecoli.fasta <span class="dt">\</span></span>
<span id="cb23-3"><a href="04-Hybrid_assembly.html#cb23-3" aria-hidden="true" tabindex="-1"></a>--frags pilon/ecoli/aln_i1.sort.bam <span class="dt">\</span></span>
<span id="cb23-4"><a href="04-Hybrid_assembly.html#cb23-4" aria-hidden="true" tabindex="-1"></a>--outdir pilon/ecoli/ <span class="dt">\</span></span>
<span id="cb23-5"><a href="04-Hybrid_assembly.html#cb23-5" aria-hidden="true" tabindex="-1"></a>--output ecoli_i1 <span class="dt">\</span></span>
<span id="cb23-6"><a href="04-Hybrid_assembly.html#cb23-6" aria-hidden="true" tabindex="-1"></a>--changes </span></code></pre></div>
<p><strong>Parameters</strong></p>
<ul>
<li><code>--genome</code> : The input genome to be polished.</li>
<li><code>--frags</code> : <code>BAM</code> containing fragment paired-end alignment using <code>bwa</code> or <code>bowtie2</code>.
<ul>
<li>There are other options for other types of Illumina reads and sequencing technologies than can be checked using <code>pilon --help | less</code>.</li>
</ul></li>
<li><code>--outdir</code> : Output directory for all output files.</li>
<li><code>--output</code> : Set the prefix for output files.</li>
<li><code>--changes</code> : This will produce a file listing the polishing changes. More info below.</li>
</ul>
<p>In the output directory there will be two files:</p>
<ol style="list-style-type: decimal">
<li><code>ecoli_i1.fasta</code> : The polished fasta file.</li>
<li><code>ecoli_i1.changes</code> : This file lists all the changes polishing carried out. There is one line per change.</li>
</ol>
<p>Let us find out how many changes <code>Pilon</code> polishing carried out:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="04-Hybrid_assembly.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> pilon/ecoli/ecoli_i1.changes</span></code></pre></div>
<p>That is a lot of changes!</p>
<p><strong>Task</strong></p>
<p>It would be good to polish the genome again. Therefore polish it once more using:</p>
<ul>
<li>The <code>Pilon</code> method like above.</li>
<li>The pilon polished assembly <code>pilon/ecoli/ecoli_i1.fasta</code> as the input for the <code>bwa</code> and <code>pilon</code> commands.</li>
<li><code>i2</code> instead of <code>i1</code> for the new files you make.</li>
</ul>
<p>Once polishing is done check the number of changes. Has the number decreased a lot?</p>
</div>
<div id="hybrid-approaches-recap" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Hybrid approaches recap</h2>
<p><img src="figures/recap.png" width="15%" style="display: block; margin: auto;" /></p>
<p>Great now you can carry out and hybrid assembly and polishing with short reads. If you produce a hybrid assembly it may also be a good idea to polish the assembly with <code>Racon</code> or <code>Pilon</code>.</p>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="03-High_coverage.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05-Appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
