---
title: "Bacterial genome assembly and annotation supplemental"
author: "Matthew R. Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
favicon: figures/NEOF_favicon.png
description: NEOF supplemental book for the Bacterial genome assembly and annotation workshop
cover-image: "figures/NEOF.png"
---

```{r cite-packages, include = FALSE}
# automatically create a bib database for R packages
# add any packages you want to cite here
knitr::write_bib(c(
  .packages(), 'bookdown', 'webexercises'
), 'packages.bib')
```

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
```

# Supplemental

```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/bact_genome_supplement.png", auto_pdf = TRUE)
```

This bookdown is a supplement to the main one. It contains some workflows and tools that are not part of the standard workflow. However, they may prove useful depending on your needs.

The sections in this supplement will cover:

- [Running the standard workflow with multiple samples](#multisampleworkflow)
- [Having too much sequencing coverage](#coverage)
- [Using long and short reads in hybrid assembly approaches](#hybrid)

Commands are in the following font, colour, and box.They should be run in the command line.

```{bash eval=FALSE}
echo "This is a command example" 
```
