# Hybrid assembly {#hybrid}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE}
knitr::include_graphics(path = "figures/hybrid_car.png", auto_pdf = TRUE)
``` 

Hybrid assembly methods can be used when you have long read (PacBio or ONT) and short read (Illumina) sequencing data for your samples. There are two main methods for this.

1. Using both short read and long read data during the assembly step (`hybridSPAdes`)
2. Assemble with the long reads and polish with the short reads (`pilon`)

## Hybrid assembly: Conda environment & directory
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/conda.png", auto_pdf = TRUE)
``` 

We will use the `hybrid` `conda` environment for our hybrid assembly methods.

Activate the environment:

```{bash eval=FALSE}
chos 8
. usehybrid
```

Before running any commands we will get the Illumina and PacBio data directories setup.

```{bash eval=FALSE}
#Change directory to bacterial_assembly directory
cd ~/bacterial_assembly/
#Make a hybrid analysis directory
mkdir hybrid_methods
#Move into the hybrid_methods directory
cd hybrid_methods
#Make directories to contain softlinks for Illumina and PacBio data
mkdir illumina_data
mkdir pacbio_data
#Make softlinks of E.coli sequencing data
ln -s /pub39/tea/nsc006/NEOF/bact_assembly/illumina_data/* illumina_data
ln -s /pub39/tea/nsc006/NEOF/bact_assembly/data/ecoli.fastq pacbio_data
```

## hybridSPAdes
```{r, fig.align = 'center',out.width= '15%', echo=FALSE}
knitr::include_graphics(path = "figures/spades.png", auto_pdf = TRUE)
``` 

A popular hybrid assembler for bacterial genomes is `hybridSPAdes`. It's use is very similar to just using Illumina data. The only difference is the addition of the `--pacbio` for PacBio reads or `--nanopore` for Oxford nanopore reads.

First we will create an output directory for `HybridSPAdes`.

```{bash eval=FALSE}
#Main output directory
mkdir hybridspades
#Ouptut directory for our sample
mkdir hybridspades/ecoli
```

Now to run `HybridSPAdes` with our Illumina and PacBio data.

```{bash eval=FALSE}
spades.py \
-1 illumina_data/ecoli_R1.fastq \
-2 illumina_data/ecoli_R2.fastq \
--pacbio pacbio_data/ecoli.fastq \
-o hybridspades/ecoli \
-t 8 -m 50
```

__Note__: This command will take about 10 minutes.

__Parameters__:

- `-1` : The forward paired-end reads.
- `-2` : The reverse paired-end reads.
- `--pacbio` : Provide PacBio reads for a hybrid assembly.
   - `--nanopore` : Used for ONT data.
- `-o` : The output directory.
- `-t` : The number of threads to be used with the process (default is 16).
- `-m` : The max RAM (in Gb) to be used for the assembly (default is 250Gb).

`SPAdes` can be very memory (RAM) hungry. This can be dangerous as using too much RAM is an easy way to crash the machine you are using. The flag `-m` is very good to prevent this . Additionally, decreasing the number of threads (`-t`) will decrease the RAM load.

Next view the output directory of this command.

```{bash eval=FALSE}
ls hybridspades/ecoli
```

You will notice a lot of output. Most of this can be deleted (especially the directories starting with `K`). The actually assembly in fasta format is the file `contigs.fasta`.

For more info on all the output please see: https://github.com/ablab/spades#spadesoutsec

## Pilon
```{r, fig.align = 'center',out.width= '15%', echo=FALSE}
knitr::include_graphics(path = "figures/pilon.png", auto_pdf = TRUE)
``` 

Another method for a hybrid approach is to carry out the standard workflow using only the long reads and then polish with short (Illumina) reads.

In this case we are going to carry out polishing using the tool `Pilon`. In theory assembling with the long reads is good to produce a very continuous assembly and then polishing with Illumina reads should drastically improve the quality and accuracy.

First we will make a directory and make a softlink of the circularised genome assembly from the standard workflow.

```{bash eval=FALSE}
#Move to main directory
#Change directory to bacterial_assembly directory
cd ~/bacterial_assembly/hybrid_methods
#Make assembly for long read assembly
mkdir long_read_assembly
#Softlink of assembly
#Using a premade file in case you don't have yours
ln -s /pub39/tea/nsc006/NEOF/bact_assembly/standard_workflow/ecoli.fasta long_read_assembly
```

The next part is to align our Illumina reads to our assembly. This is similar to the process for `Racon`. 

First we index the assembly.

```{bash eval=FALSE}
bwa index long_read_assembly/ecoli.fasta
```

Next we align our reads to the assembly we want polished. We carry this out with `bwa` which aligns Illumina reads by default.

```{bash eval=FALSE}
#Create output directory
mkdir -p pilon/ecoli
#Align the reads
bwa mem \
long_read_assembly/ecoli.fasta \
illumina_data/ecoli_R1.fastq \
illumina_data/ecoli_R2.fastq > \
pilon/ecoli/aln_i1.sam
```

__Note__: We have included `i1` in the output file to indicate  this is the first polishing iteration.

`Pilon` requires an indexed sorted `BAM` file. We will use `samtools` to produce this from our `SAM` file.

```{bash eval=FALSE}
#Create a bam file from the sam file
samtools view -Sb pilon/ecoli/aln_i1.sam > pilon/ecoli/aln_i1.bam
#Delete the old sam file
rm pilon/ecoli/aln_i1.sam
#Sort the bam file
samtools sort pilon/ecoli/aln_i1.bam -o pilon/ecoli/aln_i1.sort.bam
#delete the original bam file
rm pilon/ecoli/aln_i1.bam
#Index the sorted bam file
samtools index pilon/ecoli/aln_i1.sort.bam
```

Now we have a sorted `.bam` file and a `.bam.bai` file that contains the index information.

With this we can carry out `Pilon` polishing. `Pilon` is a `java` based program so we need to set the max memory (RAM) outside of the actual command. We'll assign the max RAM of all future java based programs within the current terminal (including `Pilon`) to 30 Gb.

```{bash eval=FALSE}
export _JAVA_OPTIONS="-Xmx30g"
```

__Yes__, the `_` before `JAVA` needs to be there. For more info on Java heap sizes check out: https://alvinalexander.com/blog/post/java/java-xmx-xms-memory-heap-size-control/

With the maximum memory set to 30Gb (the default size is too low) we can run our first iteration of `Pilon`.

```{bash eval=FALSE}
pilon \
--genome long_read_assembly/ecoli.fasta \
--frags pilon/ecoli/aln_i1.sort.bam \
--outdir pilon/ecoli/ \
--output ecoli_i1 \
--changes 
```

__Parameters__

- `--genome` : The input genome to be polished.
- `--frags` : `BAM` containing fragment paired-end alignment using `bwa` or `bowtie2`.
   - There are other options for other types of Illumina reads and sequencing technologies than can be checked using `pilon --help | less`.
- `--outdir` : Output directory for all output files.
- `--output` : Set the prefix for output files.
- `--changes` : This will produce a file listing the polishing changes. More info below.

In the output directory there will be two files:

1. `ecoli_i1.fasta` : The polished fasta file.
2. `ecoli_i1.changes` : This file lists all the changes polishing carried out. There is one line per change.

Let us find out how many changes `Pilon` polishing carried out:

```{bash eval=FALSE}
wc -l pilon/ecoli/ecoli_i1.changes
```

That is a lot of changes! 

__Task__

It would be good to polish the genome again. Therefore polish it once more using:

- The `Pilon` method like above.
- The pilon polished assembly `pilon/ecoli/ecoli_i1.fasta` as the input for the `bwa` and `pilon` commands.
- `i2` instead of `i1` for the new files you make.

Once polishing is done check the number of changes. Has the number decreased a lot?

## Hybrid approaches recap
```{r, fig.align = 'center',out.width= '15%', echo=FALSE}
knitr::include_graphics(path = "figures/recap.png", auto_pdf = TRUE)
``` 

Great now you can carry out and hybrid assembly and polishing with short reads. If you produce a hybrid assembly it may also be a good idea to polish the assembly with `Racon` or `Pilon`.
